<!doctype html>
<html class="no-js" lang="en">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><link rel="index" title="Index" href="../../../genindex.html" /><link rel="search" title="Search" href="../../../search.html" />

    <!-- Generated with Sphinx 6.1.3 and Furo 2023.03.27 -->
        <title>pertpy.data._datasets - pertpy</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo.css?digest=fad236701ea90a88636c2a8c73b44ae642ed2a53" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sg_gallery.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/override.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/sphinx_gallery.css" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #003262;
  --color-brand-content: #003262per;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  --code-font-size: var(--font-size--small);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../../index.html"><div class="brand">pertpy</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon no-toc" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../../index.html">
  
  
  
</a><form class="sidebar-search-container" method="get" action="../../../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../../usage/usage.html">Usage</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.adamson_2016_pilot.html">pertpy.data.adamson_2016_pilot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.adamson_2016_upr_epistasis.html">pertpy.data.adamson_2016_upr_epistasis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.adamson_2016_upr_perturb_seq.html">pertpy.data.adamson_2016_upr_perturb_seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.aissa_2021.html">pertpy.data.aissa_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.bhattacherjee.html">pertpy.data.bhattacherjee</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.burczynski_crohn.html">pertpy.data.burczynski_crohn</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.chang_2021.html">pertpy.data.chang_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.datlinger_2017.html">pertpy.data.datlinger_2017</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.datlinger_2021.html">pertpy.data.datlinger_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dialogue_example.html">pertpy.data.dialogue_example</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dixit_2016.html">pertpy.data.dixit_2016</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dixit_2016_raw.html">pertpy.data.dixit_2016_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dixit_2016_scperturb.html">pertpy.data.dixit_2016_scperturb</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021.html">pertpy.data.frangieh_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021_protein.html">pertpy.data.frangieh_2021_protein</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021_raw.html">pertpy.data.frangieh_2021_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021_rna.html">pertpy.data.frangieh_2021_rna</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.gasperini_2019_atscale.html">pertpy.data.gasperini_2019_atscale</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.gasperini_2019_highmoi.html">pertpy.data.gasperini_2019_highmoi</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.gasperini_2019_lowmoi.html">pertpy.data.gasperini_2019_lowmoi</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.gehring_2019.html">pertpy.data.gehring_2019</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.haber_2017_regions.html">pertpy.data.haber_2017_regions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.kang_2018.html">pertpy.data.kang_2018</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.mcfarland_2020.html">pertpy.data.mcfarland_2020</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.norman_2019.html">pertpy.data.norman_2019</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.norman_2019_raw.html">pertpy.data.norman_2019_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.papalexi_2021.html">pertpy.data.papalexi_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.replogle_2022_k562_essential.html">pertpy.data.replogle_2022_k562_essential</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.replogle_2022_k562_gwps.html">pertpy.data.replogle_2022_k562_gwps</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.replogle_2022_rpe1.html">pertpy.data.replogle_2022_rpe1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.sc_sim_augur.html">pertpy.data.sc_sim_augur</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.schiebinger_2019_16day.html">pertpy.data.schiebinger_2019_16day</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.schiebinger_2019_18day.html">pertpy.data.schiebinger_2019_18day</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr8.html">pertpy.data.schraivogel_2020_tap_screen_chr8</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr11.html">pertpy.data.schraivogel_2020_tap_screen_chr11</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.sciplex3_raw.html">pertpy.data.sciplex3_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.shifrut_2018.html">pertpy.data.shifrut_2018</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.smillie.html">pertpy.data.smillie</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex2.html">pertpy.data.srivatsan_2020_sciplex2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex3.html">pertpy.data.srivatsan_2020_sciplex3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex4.html">pertpy.data.srivatsan_2020_sciplex4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.stephenson_2021_subsampled.html">pertpy.data.stephenson_2021_subsampled</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.tian_2019_day7neuron.html">pertpy.data.tian_2019_day7neuron</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.tian_2019_ipsc.html">pertpy.data.tian_2019_ipsc</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.tian_2021_crispra.html">pertpy.data.tian_2021_crispra</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.tian_2021_crispri.html">pertpy.data.tian_2021_crispri</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.weinreb_2020.html">pertpy.data.weinreb_2020</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.xie_2017.html">pertpy.data.xie_2017</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.zhao_2021.html">pertpy.data.zhao_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/preprocessing/pertpy.preprocessing.GuideAssignment.html">pertpy.preprocessing.GuideAssignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Augur.html">pertpy.tools.Augur</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Mixscape.html">pertpy.tools.Mixscape</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Milo.html">pertpy.tools.Milo</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Sccoda.html">pertpy.tools.Sccoda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Tasccoda.html">pertpy.tools.Tasccoda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Dialogue.html">pertpy.tools.Dialogue</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Distance.html">pertpy.tools.Distance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.DistanceTest.html">pertpy.tools.DistanceTest</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.CellLineMetaData.html">pertpy.tools.CellLineMetaData</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.SCGEN.html">pertpy.tools.SCGEN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.kernel_pca.html">pertpy.tools.kernel_pca</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.guide.html">pertpy.plot.guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.dp_scatter.html">pertpy.plot.ag.dp_scatter</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.important_features.html">pertpy.plot.ag.important_features</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.lollipop.html">pertpy.plot.ag.lollipop</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.scatterplot.html">pertpy.plot.ag.scatterplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.violin.html">pertpy.plot.ms.violin</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.perturbscore.html">pertpy.plot.ms.perturbscore</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.heatmap.html">pertpy.plot.ms.heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.barplot.html">pertpy.plot.ms.barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.lda.html">pertpy.plot.ms.lda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.milo.nhood_graph.html">pertpy.plot.milo.nhood_graph</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.milo.nhood.html">pertpy.plot.milo.nhood</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.milo.da_beeswarm.html">pertpy.plot.milo.da_beeswarm</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.milo.nhood_counts_by_cond.html">pertpy.plot.milo.nhood_counts_by_cond</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.stacked_barplot.html">pertpy.plot.coda.stacked_barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.effects_barplot.html">pertpy.plot.coda.effects_barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.boxplots.html">pertpy.plot.coda.boxplots</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.rel_abundance_dispersion_plot.html">pertpy.plot.coda.rel_abundance_dispersion_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.draw_tree.html">pertpy.plot.coda.draw_tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.draw_effects.html">pertpy.plot.coda.draw_effects</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.coda.effects_umap.html">pertpy.plot.coda.effects_umap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.scg.reg_mean_plot.html">pertpy.plot.scg.reg_mean_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.scg.reg_var_plot.html">pertpy.plot.scg.reg_var_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.scg.binary_classifier.html">pertpy.plot.scg.binary_classifier</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">Contributor Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../code_of_conduct.html">Contributor Covenant Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/theislab/pertpy/discussions">Discussions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../authors.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../references.html">References</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon no-toc" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <h1>Source code for pertpy.data._datasets</h1><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="kn">import</span> <span class="nn">muon</span> <span class="k">as</span> <span class="nn">mu</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">from</span> <span class="nn">anndata</span> <span class="kn">import</span> <span class="n">AnnData</span>
<span class="kn">from</span> <span class="nn">mudata</span> <span class="kn">import</span> <span class="n">MuData</span>
<span class="kn">from</span> <span class="nn">scanpy</span> <span class="kn">import</span> <span class="n">settings</span>

<span class="kn">from</span> <span class="nn">pertpy.data._dataloader</span> <span class="kn">import</span> <span class="n">_download</span>


<div class="viewcode-block" id="burczynski_crohn"><a class="viewcode-back" href="../../../usage/data/pertpy.data.burczynski_crohn.html#pertpy.data.burczynski_crohn">[docs]</a><span class="k">def</span> <span class="nf">burczynski_crohn</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Bulk data with conditions ulcerative colitis (UC) and Crohn&#39;s disease (CD).</span>

<span class="sd">    The study assesses transcriptional profiles in peripheral blood mononuclear</span>
<span class="sd">    cells from 42 healthy individuals, 59 CD patients, and 26 UC patients by</span>
<span class="sd">    hybridization to microarrays interrogating more than 22,000 sequences.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Burczynski et al., &quot;Molecular classification of Crohn&#39;s disease and</span>
<span class="sd">        ulcerative colitis patients using transcriptional profiles in peripheral blood mononuclear cells&quot;</span>
<span class="sd">        J Mol Diagn 8, 51 (2006). PMID:16436634.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of the Burczynski dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">return</span> <span class="n">sc</span><span class="o">.</span><span class="n">datasets</span><span class="o">.</span><span class="n">burczynski06</span><span class="p">()</span></div>


<div class="viewcode-block" id="papalexi_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.papalexi_2021.html#pertpy.data.papalexi_2021">[docs]</a><span class="k">def</span> <span class="nf">papalexi_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">MuData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;ECCITE-seq dataset of 11 gRNAs generated from stimulated THP-1 cell line.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Papalexi, E., Mimitou, E.P., Butler, A.W. et al. Characterizing the molecular regulation</span>
<span class="sd">        of inhibitory immune checkpoints with multimodal single-cell screens.</span>
<span class="sd">        Nat Genet 53, 322–331 (2021). https://doi.org/10.1038/s41588-021-00778-2</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of the ECCITE-seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;papalexi_2021.h5mu&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/36509460&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">mudata</span> <span class="o">=</span> <span class="n">mu</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">mudata</span> <span class="o">=</span> <span class="n">mu</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">mudata</span></div>


<div class="viewcode-block" id="sc_sim_augur"><a class="viewcode-back" href="../../../usage/data/pertpy.data.sc_sim_augur.html#pertpy.data.sc_sim_augur">[docs]</a><span class="k">def</span> <span class="nf">sc_sim_augur</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Simulated test dataset used the usage example for the Augur.</span>

<span class="sd">    https://github.com/neurorestore/Augur</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a simulated single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;sc_sim_augur.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/31645886&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="bhattacherjee"><a class="viewcode-back" href="../../../usage/data/pertpy.data.bhattacherjee.html#pertpy.data.bhattacherjee">[docs]</a><span class="k">def</span> <span class="nf">bhattacherjee</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Processed single-cell data PFC adult mice under cocaine self-administration.</span>

<span class="sd">    Adult mice were subject to cocaine self-administration, samples were</span>
<span class="sd">    collected a three time points: Maintance, 48h after cocaine withdrawal and</span>
<span class="sd">    15 days after cocaine withdrawal.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Bhattacherjee A, Djekidel MN, Chen R, Chen W, Tuesta LM, Zhang Y. Cell</span>
<span class="sd">        type-specific transcriptional programs in mouse prefrontal cortex during</span>
<span class="sd">        adolescence and addiction. Nat Commun. 2019 Sep 13;10(1):4169.</span>
<span class="sd">        doi: 10.1038/s41467-019-12054-3. PMID: 31519873; PMCID: PMC6744514.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;bhattacherjee_rna.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34526528&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="sciplex3_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.sciplex3_raw.html#pertpy.data.sciplex3_raw">[docs]</a><span class="k">def</span> <span class="nf">sciplex3_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Raw sciplex3 perturbation dataset curated for perturbation modeling.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Srivatsan SR, McFaline-Figueroa JL, Ramani V, Saunders L, Cao J, Packer J,</span>
<span class="sd">        Pliner HA, Jackson DL, Daza RM, Christiansen L, Zhang F, Steemers F,</span>
<span class="sd">        Shendure J, Trapnell C. Massively multiplex chemical transcriptomics at</span>
<span class="sd">        single-cell resolution. Science. 2020 Jan 3;367(6473):45-51.</span>
<span class="sd">        doi: 10.1126/science.aax6234. Epub 2019 Dec 5. PMID: 31806696; PMCID: PMC7289078.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;sciplex3.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/33979517&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="smillie"><a class="viewcode-back" href="../../../usage/data/pertpy.data.smillie.html#pertpy.data.smillie">[docs]</a><span class="k">def</span> <span class="nf">smillie</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;scRNA-seq data of the small intestine of mice under Ulcerative Colitis.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Smillie, Christopher S et al. “Intra- and Inter-cellular Rewiring of the Human Colon during Ulcerative Colitis.”</span>
<span class="sd">        Cell vol. 178,3 (2019): 714-730.e22. doi:10.1016/j.cell.2019.06.029</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;smillie.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/38648585&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021.html#pertpy.data.frangieh_2021">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Processed perturb-CITE-seq data with multi-modal RNA and protein single-cell profiling.</span>

<span class="sd">    We profiled RNA and 20 surface proteins in over 218,000 cells under ~750 perturbations,</span>
<span class="sd">    chosen by their membership in an immune evasion program that is associated with</span>
<span class="sd">    immunotherapy resistance in patients.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Frangieh, C.J., Melms, J.C., Thakore, P.I. et al. Multimodal pooled Perturb-CITE-seq</span>
<span class="sd">        screens in patient models define mechanisms of cancer immune evasion.</span>
<span class="sd">        Nat Genet 53, 332–341 (2021). https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a Perturb-CITE-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34013717&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021_raw.html#pertpy.data.frangieh_2021_raw">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Raw Perturb-CITE-seq data with multi-modal RNA and protein single-cell profiling readout.</span>

<span class="sd">    We profiled RNA and 20 surface proteins in over 218,000 cells under ~750 perturbations,</span>
<span class="sd">    chosen by their membership in an immune evasion program that is associated with</span>
<span class="sd">    immunotherapy resistance in patients.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Frangieh, C.J., Melms, J.C., Thakore, P.I. et al. Multimodal pooled Perturb-CITE-seq</span>
<span class="sd">        screens in patient models define mechanisms of cancer immune evasion.</span>
<span class="sd">        Nat Genet 53, 332–341 (2021). https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of raw Perturb-CITE-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34012565&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dixit_2016_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dixit_2016_raw.html#pertpy.data.dixit_2016_raw">[docs]</a><span class="k">def</span> <span class="nf">dixit_2016_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Perturb-seq: scRNA-seq with pooled CRISPR-KO perturbations.</span>

<span class="sd">    scRNA-seq with pooled CRISPR-KO perturbations in 200,000 cells across six screens</span>
<span class="sd">    unstimulated BMDC, BMDC stimulated at 3hr, TFs in K562 at 7 and 13 days post transduction,</span>
<span class="sd">    and 13 days at a higher MOI of perturbations.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Dixit A, Parnas O, Li B, Chen J et al. Perturb-Seq: Dissecting Molecular Circuits with</span>
<span class="sd">        Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.</span>
<span class="sd">        Cell 2016 Dec 15;167(7):1853-1866.e17. DOI:https://doi.org/10.1016/j.cell.2016.11.038</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of raw Perturb-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dixit_2016_raw.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34012565&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dixit_2016"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dixit_2016.html#pertpy.data.dixit_2016">[docs]</a><span class="k">def</span> <span class="nf">dixit_2016</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Perturb-seq: scRNA-seq with pooled CRISPR-KO perturbations.</span>

<span class="sd">    scRNA-seq with pooled CRISPR-KO perturbations in 200,000 cells across six screens</span>
<span class="sd">    unstimulated BMDC, BMDC stimulated at 3hr, TFs in K562 at 7 and 13 days post trasnduction,</span>
<span class="sd">    and 13 days at a higher MOI of perturbations.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Dixit A, Parnas O, Li B, Chen J et al. Perturb-Seq: Dissecting Molecular Circuits with</span>
<span class="sd">        Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.</span>
<span class="sd">        Cell 2016 Dec 15;167(7):1853-1866.e17. DOI:https://doi.org/10.1016/j.cell.2016.11.038</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of Perturb-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dixit_2016.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34014608&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="norman_2019"><a class="viewcode-back" href="../../../usage/data/pertpy.data.norman_2019.html#pertpy.data.norman_2019">[docs]</a><span class="k">def</span> <span class="nf">norman_2019</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Processed single-cell, pooled CRISPR screening.</span>

<span class="sd">    Single-cell, pooled CRISPR screening experiment comparing the transcriptional effects of</span>
<span class="sd">    overexpressing genes alone or in combination</span>

<span class="sd">    Reference:</span>
<span class="sd">        Norman, Thomas M et al. “Exploring genetic interaction manifolds constructed from rich</span>
<span class="sd">        single-cell phenotypes.” Science (New York, N.Y.) vol. 365,6455 (2019): 786-793.</span>
<span class="sd">        doi:10.1126/science.aax4438</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of sc pooled CRISPR screening</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;norman_2019.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34027562&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="norman_2019_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.norman_2019_raw.html#pertpy.data.norman_2019_raw">[docs]</a><span class="k">def</span> <span class="nf">norman_2019_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Raw single-cell, pooled CRISPR screening.</span>

<span class="sd">    Single-cell, pooled CRISPR screening experiment comparing the transcriptional effects of</span>
<span class="sd">    overexpressing genes alone or in combination</span>

<span class="sd">    Reference:</span>
<span class="sd">        Norman, Thomas M et al. “Exploring genetic interaction manifolds constructed from rich</span>
<span class="sd">        single-cell phenotypes.” Science (New York, N.Y.) vol. 365,6455 (2019): 786-793.</span>
<span class="sd">        doi:10.1126/science.aax4438</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of sc pooled CRISPR screening</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;norman_2019_raw.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34002548&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dialogue_example"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dialogue_example.html#pertpy.data.dialogue_example">[docs]</a><span class="k">def</span> <span class="nf">dialogue_example</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Example dataset used to feature DIALOGUE.</span>

<span class="sd">    https://github.com/livnatje/DIALOGUE/wiki/Example</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dialogue_example.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34490714&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<span class="k">def</span> <span class="nf">distance_example_data</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Example dataset used to feature distances and distance_tests.</span>

<span class="sd">    Processed and subsetted version of original Perturb-seq dataset by Dixit et al.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;distances_example_data.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/39561379&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span>


<div class="viewcode-block" id="kang_2018"><a class="viewcode-back" href="../../../usage/data/pertpy.data.kang_2018.html#pertpy.data.kang_2018">[docs]</a><span class="k">def</span> <span class="nf">kang_2018</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Processed multiplexing droplet-based single cell RNA-sequencing using genetic barcodes.</span>

<span class="sd">    HiSeq 2500 data for sequencing of PBMCs from SLE patients and 2 controls. We collected 1M cells</span>
<span class="sd">    each from frozen PBMC samples that were Ficoll isolated and prepared using the 10x Single Cell</span>
<span class="sd">    instrument according to standard protocol. Samples A, B, and C were prepared on the instrument</span>
<span class="sd">    directly following thaw, while samples 2.1 and 2.2 were cultured for 6 hours with (B) or</span>
<span class="sd">    without (A) IFN-beta stimulation prior to loading on the 10x Single Cell instrument.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Kang, H., Subramaniam, M., Targ, S. et al. Multiplexed droplet single-cell RNA-sequencing</span>
<span class="sd">        using natural genetic variation.</span>
<span class="sd">        Nat Biotechnol 36, 89–94 (2018). https://doi.org/10.1038/nbt.4042</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of droplet-based single cell RNA-sequencing</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;kang_2018.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34464122&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="stephenson_2021_subsampled"><a class="viewcode-back" href="../../../usage/data/pertpy.data.stephenson_2021_subsampled.html#pertpy.data.stephenson_2021_subsampled">[docs]</a><span class="k">def</span> <span class="nf">stephenson_2021_subsampled</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Processed 10X 5&#39; scRNA-seq data from PBMC of COVID-19 patients and healthy donors</span>

<span class="sd">    The study profiled peripheral blood mononuclear cells from 90 COVID-19 patients with different disease severity and 23 healthy control donors.</span>
<span class="sd">    Here the dataset was downsampled to approx. 500 cells per donor and cells were mapped to a reference atlas of healthy PBMCs from 12 studies</span>
<span class="sd">    with scArches.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Stephenson, E., Reynolds, G., Botting, R. A., et al. (2021).</span>
<span class="sd">        Single-cell multi-omics analysis of the immune response in COVID-19.</span>
<span class="sd">        Nature Medicine, 27(5). https://doi.org/10.1038/s41591-021-01329-2</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scRNA-seq profiles</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;stephenson_2021_subsampled.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/38171703&quot;</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="haber_2017_regions"><a class="viewcode-back" href="../../../usage/data/pertpy.data.haber_2017_regions.html#pertpy.data.haber_2017_regions">[docs]</a><span class="k">def</span> <span class="nf">haber_2017_regions</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Raw single-cell, pooled CRISPR screening.</span>

<span class="sd">    Epithelial cells from the small intestine and organoids of mice.</span>
<span class="sd">    Some of the cells were also subject to Salmonella or Heligmosomoides polygyrus infection (day 3 and day 10).</span>

<span class="sd">    Reference:</span>
<span class="sd">        Haber, Adam L. et al. “A single-cell survey of the small intestinal epithelium” Nature vol. 551 (2017): 333-339</span>
<span class="sd">        doi:10.1038/nature24489</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;haber_2017_regions.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/38169900&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="adamson_2016_pilot"><a class="viewcode-back" href="../../../usage/data/pertpy.data.adamson_2016_pilot.html#pertpy.data.adamson_2016_pilot">[docs]</a><span class="k">def</span> <span class="nf">adamson_2016_pilot</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;6000 chronic myeloid leukemia (K562) cells carrying 8 distinct GBCs.</span>

<span class="sd">    In a pilot experiment, single-cell RNA-seq was performed on a pool of individually transduced chronic</span>
<span class="sd">    myeloid leukemia cells (K562) carrying 8 distinct guide barcodes, analyzing ∼6,000 cells total.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.sciencedirect.com/science/article/pii/S0092867416316609?via%3Dihub</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;adamson_2016_pilot.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/AdamsonWeissman2016_GSM2406675_10X001.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="adamson_2016_upr_epistasis"><a class="viewcode-back" href="../../../usage/data/pertpy.data.adamson_2016_upr_epistasis.html#pertpy.data.adamson_2016_upr_epistasis">[docs]</a><span class="k">def</span> <span class="nf">adamson_2016_upr_epistasis</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;15000 K562 cells with UPR sensor genes knocked out and treated with thapsigargin.</span>

<span class="sd">    In UPR epistasis experiment, Perturb-seq was applied to explore the branches of</span>
<span class="sd">    the mammalian UPR. Using the three-guide Perturb-seq vector, sgRNAs targeting each</span>
<span class="sd">    UPR sensor gene were introduced into K562 cells with dCas9-KRAB. Transduced cells</span>
<span class="sd">    were then pooled, sorted for vector delivery, and after 5 days of total growth,</span>
<span class="sd">    treated with thapsigargin. Control cells were treated with DMSO. Transcriptomes</span>
<span class="sd">    of ∼15,000 cells were sequenced.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.sciencedirect.com/science/article/pii/S0092867416316609?via%3Dihub</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb preparedsingle-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;adamson_2016_upr_epistasis.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/AdamsonWeissman2016_GSM2406677_10X005.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="adamson_2016_upr_perturb_seq"><a class="viewcode-back" href="../../../usage/data/pertpy.data.adamson_2016_upr_perturb_seq.html#pertpy.data.adamson_2016_upr_perturb_seq">[docs]</a><span class="k">def</span> <span class="nf">adamson_2016_upr_perturb_seq</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 65000 cells that were subject to 91 sgRNAs targeting 82 genes.</span>

<span class="sd">    In UPR Perturb-seq experiment, Perturb-seq was applied to a small CRISPRi library</span>
<span class="sd">    of 91 sgRNAs targeting 82 genes. sgRNAs were delivered via pooled transduction</span>
<span class="sd">    using a mixture of separately prepared lentiviruses, and ∼65,000 transcriptomes</span>
<span class="sd">    were collected in one large pooled experiment.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.sciencedirect.com/science/article/pii/S0092867416316609?via%3Dihub</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;adamson_2016_upr_perturb_seq.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/AdamsonWeissman2016_GSM2406681_10X010.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="aissa_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.aissa_2021.html#pertpy.data.aissa_2021">[docs]</a><span class="k">def</span> <span class="nf">aissa_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 848 P99 cells subject to consecutive erlotinib and 756 control cells.</span>

<span class="sd">    In this study 848 PC9 cells subjected to consecutive erlotinib treatment (for</span>
<span class="sd">    1, 2, 4, 9, and 11 days) and 756 control cells were analysed using Drop-seq.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.nature.com/articles/s41467-021-21884-z</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;aissa_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/AissaBenevolenskaya2021.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="chang_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.chang_2021.html#pertpy.data.chang_2021">[docs]</a><span class="k">def</span> <span class="nf">chang_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 5 different cell lines that were induced with a unique TraCe-seq barcode.</span>

<span class="sd">    TraCe-seq is a method that captures at clonal resolution the origin, fate and</span>
<span class="sd">    differential early adaptive transcriptional programs of cells in a complex</span>
<span class="sd">    population in response to distinct treatments. Here, a unique TraCe-seq barcode was</span>
<span class="sd">    transduced into five different cell lines (PC9, MCF-10A, MDA-MB-231, NCI-H358 and NCI-H1373).</span>
<span class="sd">    Transduced cells were selected with puromycin only, mixed together and profiled by 10x scRNA-seq.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.nature.com/articles/s41587-021-01005-3</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;chang_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ChangYe2021.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="datlinger_2017"><a class="viewcode-back" href="../../../usage/data/pertpy.data.datlinger_2017.html#pertpy.data.datlinger_2017">[docs]</a><span class="k">def</span> <span class="nf">datlinger_2017</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptpmics of 5905 Jurkat cells induced with anti-CD3 and anti-CD28 antibodies.</span>

<span class="sd">    For CROP-seq, Jurkat cells were transduced with a gRNA library targeting high-level</span>
<span class="sd">    regulators of T cell receptor signaling and a set of transcription factors. After 10</span>
<span class="sd">    days of antibiotic selection and expansion, cells were stimulated with anti-CD3 and</span>
<span class="sd">    anti-CD28 antibodies or left untreated. Both conditions were analyzed using CROP-seq,</span>
<span class="sd">    measuring TCR activation for each gene knockout. The dataset comprises 5,905 high-quality</span>
<span class="sd">    single-cell transcriptomes with uniquely assigned gRNAs.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://www.nature.com/articles/nmeth.4177</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;datlinger_2017.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/DatlingerBock2017.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="datlinger_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.datlinger_2021.html#pertpy.data.datlinger_2021">[docs]</a><span class="k">def</span> <span class="nf">datlinger_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 151788 nuclei of four cell lines.</span>

<span class="sd">    A large-scale scifi-RNA-seq experiment was performed with 383,000 nuclei loaded into</span>
<span class="sd">    a single microfluidic channel of the Chromium system. Four human cell lines (HEK293T,</span>
<span class="sd">    Jurkat, K562 and NALM-6) were combined in an equal mixture, and technical replicates of</span>
<span class="sd">    each cell line with different preindexing (round1) barcodes were marked. This experiment</span>
<span class="sd">    resulted in 151,788 single-cell transcriptomes passing quality control.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41592-021-01153-z</span>

<span class="sd">     Returns:</span>
<span class="sd">         :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;datlinger_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/DatlingerBock2021.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dixit_2016_scperturb"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dixit_2016_scperturb.html#pertpy.data.dixit_2016_scperturb">[docs]</a><span class="k">def</span> <span class="nf">dixit_2016_scperturb</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 200000 bone marrow-derived dendritic K562 cells.</span>

<span class="sd">    Six Perturb-seq experiments were performed, analyzing 200,000 cells. In bone</span>
<span class="sd">    marrow-derived dendritic cells (BMDCs), 24 transcription factors (TFs) were targeted</span>
<span class="sd">    and the effects pre-stimulation (0 hr) and at 3 hr post-lipopolysaccharide (LPS) were</span>
<span class="sd">    measured. In K562 cells, 14 TFs and 10 cell-cycle regulators were targeted in separate</span>
<span class="sd">    pooled experiments. For K562 TFs, experiments were performed using lower and higher MOI</span>
<span class="sd">    and at two time points. Reference scRNA-seq data from unperturbed cells were collected separately.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2016.11.038</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dixit_2016_scperturb.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/DixitRegev2016.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021_protein"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021_protein.html#pertpy.data.frangieh_2021_protein">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021_protein</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;CITE-seq data of 218000 cells under 750 perturbations (only the surface protein data).</span>

<span class="sd">    Perturb-CITE-seq was developed for pooled CRISPR perturbation screens with multi-modal</span>
<span class="sd">    RNA and protein single-cell profiling readout and applied to screen patient-derived</span>
<span class="sd">    autologous melanoma and tumor infiltrating lymphocyte (TIL) co-cultures. RNA and 20</span>
<span class="sd">    surface proteins were profiled in over 218,000 cells under ~750 perturbations, chosen</span>
<span class="sd">    by their membership in an immune evasion program that is associated with immunotherapy</span>
<span class="sd">    resistance in patients.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021_protein.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/FrangiehIzar2021_protein.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021_rna"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021_rna.html#pertpy.data.frangieh_2021_rna">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021_rna</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;CITE-seq data of 218000 cells under 750 perturbations (only the transcriptomics data).</span>

<span class="sd">    Perturb-CITE-seq was developed for pooled CRISPR perturbation screens with multi-modal</span>
<span class="sd">    RNA and protein single-cell profiling readout and applied to screen patient-derived</span>
<span class="sd">    autologous melanoma and tumor infiltrating lymphocyte (TIL) co-cultures. RNA and 20</span>
<span class="sd">    surface proteins were profiled in over 218,000 cells under ~750 perturbations, chosen</span>
<span class="sd">    by their membership in an immune evasion program that is associated with immunotherapy</span>
<span class="sd">    resistance in patients.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021_rna.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/FrangiehIzar2021_RNA.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="gasperini_2019_atscale"><a class="viewcode-back" href="../../../usage/data/pertpy.data.gasperini_2019_atscale.html#pertpy.data.gasperini_2019_atscale">[docs]</a><span class="k">def</span> <span class="nf">gasperini_2019_atscale</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 254974 cells of chronic K562 cells with CRISPRi perturbations.</span>

<span class="sd">    Across two experiments, the authors used dCas9-KRAB to perturb 5,920 candidate enhancers</span>
<span class="sd">    with no strong a priori hypothesis as to their target gene(s) in 254,974 cells of chronic</span>
<span class="sd">    myelogenous leukemia cell line K562, with CRISPRi as the mode of perturbation.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2018.11.029</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;gasperini_2019_atscale.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/GasperiniShendure2019_atscale.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="gasperini_2019_highmoi"><a class="viewcode-back" href="../../../usage/data/pertpy.data.gasperini_2019_highmoi.html#pertpy.data.gasperini_2019_highmoi">[docs]</a><span class="k">def</span> <span class="nf">gasperini_2019_highmoi</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;K562 perturbed cells with 1119 candidate enhancers (only the high MOI part).</span>

<span class="sd">    The authors used dCas9-KRAB to perturb 1,119 candidate enhancers with no strong a priori</span>
<span class="sd">    hypothesis as to their target gene(s) in the chronic myelogenous leukemia cell line</span>
<span class="sd">    K562, with CRISPRi as the mode of perturbation, where K562 cells were transduced</span>
<span class="sd">    at a high MOI (pilot library MOI = ∼15).</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2018.11.029</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;gasperini_2019_highmoi.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/GasperiniShendure2019_highMOI.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="gasperini_2019_lowmoi"><a class="viewcode-back" href="../../../usage/data/pertpy.data.gasperini_2019_lowmoi.html#pertpy.data.gasperini_2019_lowmoi">[docs]</a><span class="k">def</span> <span class="nf">gasperini_2019_lowmoi</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;K562 perturbed cells with 1119 candidate enhancers (only the low MOI part).</span>

<span class="sd">    The authors used dCas9-KRAB to perturb 1,119 candidate enhancers with no strong a priori</span>
<span class="sd">    hypothesis as to their target gene(s) in chronic myelogenous leukemia cell line K562,</span>
<span class="sd">    with CRISPRi as the mode of perturbation, where K562 cells were transduced at a</span>
<span class="sd">    low MOI (pilot library MOI = ∼1).</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2018.11.029</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;gasperini_2019_lowmoi.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/GasperiniShendure2019_lowMOI.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="gehring_2019"><a class="viewcode-back" href="../../../usage/data/pertpy.data.gehring_2019.html#pertpy.data.gehring_2019">[docs]</a><span class="k">def</span> <span class="nf">gehring_2019</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;96-plex perturbation experiment on live mouse neural stem cells.</span>

<span class="sd">    In this study, a 96-plex perturbation experiment was conducted on live mouse neural</span>
<span class="sd">    stem cells (NSCs), consisting of a pair of drug-triples with 4 drugs in total at 3</span>
<span class="sd">    or 4 different concentractions.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41587-019-0372-z</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a scPerturb prepared single-cell dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;gehring_2019.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/GehringPachter2019.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="mcfarland_2020"><a class="viewcode-back" href="../../../usage/data/pertpy.data.mcfarland_2020.html#pertpy.data.mcfarland_2020">[docs]</a><span class="k">def</span> <span class="nf">mcfarland_2020</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Response of various cell lines to a range of different drugs and CRISPRi perturbations.</span>

<span class="sd">    Here, the authors developed MIX-Seq, a method for multiplexed transcriptional profiling</span>
<span class="sd">    of post-perturbation responses across many cell contexts, using scRNA-seq applied to</span>
<span class="sd">    co-treated pools of cancer cell lines. The responses of pools of 24–99 cell lines to</span>
<span class="sd">    a range of different drugs were profiled, as well as to CRISPR perturbations.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41467-020-17440-w</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;mcfarland_2020.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/McFarlandTsherniak2020.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="replogle_2022_k562_essential"><a class="viewcode-back" href="../../../usage/data/pertpy.data.replogle_2022_k562_essential.html#pertpy.data.replogle_2022_k562_essential">[docs]</a><span class="k">def</span> <span class="nf">replogle_2022_k562_essential</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;K562 cells transduced with CRISPRi (day 7 after transduction).</span>

<span class="sd">    For day 6 essential-scale experiment in chronic myeloid leukemia (CML) (K562) cell</span>
<span class="sd">    lines, library lentivirus was packaged into lentivirus in 293T cells and empirically</span>
<span class="sd">    measured in K562 cells to obtain viral titers. CRISPRi K562 cells were transduced</span>
<span class="sd">    and 20Q1 Cancer Dependency Map common essential genes were targeted at day 7 after transduction.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2022.05.013</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;replogle_2022_k562_essential.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ReplogleWeissman2022_K562_essential.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="replogle_2022_k562_gwps"><a class="viewcode-back" href="../../../usage/data/pertpy.data.replogle_2022_k562_gwps.html#pertpy.data.replogle_2022_k562_gwps">[docs]</a><span class="k">def</span> <span class="nf">replogle_2022_k562_gwps</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;K562 cells transduced with CRISPRi (day 8 after transcduction).</span>

<span class="sd">    Here, the authors used a compact, multiplexed CRISPR interference (CRISPRi) library</span>
<span class="sd">    to assay thousands of loss-of-function genetic perturbations with single-cell RNA sequencing</span>
<span class="sd">    in chronic myeloid leukemia (CML) (K562) cell lines. For the K562 day 8 genome-scale</span>
<span class="sd">    Perturb-seq experiment, library lentivirus was packaged into lentivirus in 293T cells and</span>
<span class="sd">    empirically measured in K562 cells to obtain viral titers. CRISPRi K562 cells were transduced</span>
<span class="sd">    and all expressed genes were targeted at day 8 after transduction</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2022.05.013</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;replogle_2022_k562_gwps.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ReplogleWeissman2022_K562_gwps.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="replogle_2022_rpe1"><a class="viewcode-back" href="../../../usage/data/pertpy.data.replogle_2022_rpe1.html#pertpy.data.replogle_2022_rpe1">[docs]</a><span class="k">def</span> <span class="nf">replogle_2022_rpe1</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;RPE1 cells transduced with CRISPRi (day 7 after transcduction).</span>

<span class="sd">    For day 7 essential-scale Perturb-seq experiment in retinal pigment epithelial (RPE1)</span>
<span class="sd">    cell lines, library lentivirus was packaged into lentivirus in 293T cells and</span>
<span class="sd">    empirically measured in RPE1 cells to obtain viral titers. CRISPRi RPE1 cells expressing</span>
<span class="sd">    ZIM3 KRAB-dCas9-P2A-BFP were transduced. 20Q1 Cancer Dependency Map common essential</span>
<span class="sd">    genes were targeted at day 7 after transduction.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2022.05.013</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;replogle_2022_rpe1.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ReplogleWeissman2022_rpe1.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="schiebinger_2019_16day"><a class="viewcode-back" href="../../../usage/data/pertpy.data.schiebinger_2019_16day.html#pertpy.data.schiebinger_2019_16day">[docs]</a><span class="k">def</span> <span class="nf">schiebinger_2019_16day</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomes of 65781 iPSC cells collected over 10 time points in 2i or serum conditions (16-day time course).</span>

<span class="sd">    Samples were collected from established iPSC lines reprogrammed from the reprogramming mouse embryonic</span>
<span class="sd">    fibroblasts (MEFs), maintained in either 2i or serum conditions, at 10 time points across 16 days.</span>
<span class="sd">    Overall, 68,339 cells were profiled to an average depth of 38,462 reads per cell. After discarding</span>
<span class="sd">    cells with less than 1,000 genes detected, 65,781 cells were obtained, with a median of 2,398 genes</span>
<span class="sd">    and 7,387 unique transcripts per cell.</span>


<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2019.01.006</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;schiebinger_2019_16day.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SchiebingerLander2019_GSE106340.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="schiebinger_2019_18day"><a class="viewcode-back" href="../../../usage/data/pertpy.data.schiebinger_2019_18day.html#pertpy.data.schiebinger_2019_18day">[docs]</a><span class="k">def</span> <span class="nf">schiebinger_2019_18day</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomes of 259155 iPSC cells collected over 39 time points in 2i or serum conditions (18-day time course).</span>

<span class="sd">    Samples were collected from established iPSC lines reprogrammed from the reprogramming mouse embryonic</span>
<span class="sd">    fibroblasts (MEFs), maintained in either 2i or serum conditions, over 39 time points separated by</span>
<span class="sd">    ∼12 hours across an 18-day time course (and every 6 hours between days 8 and 9). Overall, 259,155 cells</span>
<span class="sd">    were profiled.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2019.01.006</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;Schiebinger_2019_18day.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SchiebingerLander2019_GSE115943.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="schraivogel_2020_tap_screen_chr11"><a class="viewcode-back" href="../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr11.html#pertpy.data.schraivogel_2020_tap_screen_chr11">[docs]</a><span class="k">def</span> <span class="nf">schraivogel_2020_tap_screen_chr11</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;TAP-seq applied to K562 cells (only chromosome 11).</span>

<span class="sd">    TAP-seq was applied to generate perturbation-based enhancer–target gene maps in K562 cells.</span>
<span class="sd">    They perturbed all 1,778 putatively active enhancers predicted on the basis of ENCODE data</span>
<span class="sd">    in two regions on chromosome 8 and 11, and identified effects on expressed protein-coding genes</span>
<span class="sd">    within the same regions. Thus, in each cell, 68 (chromosome 8) or 79 (chromosome 11) target genes were measured.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41592-020-0837-5</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;schraivogel_2020_tap_screen_chr11.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SchraivogelSteinmetz2020_TAP_SCREEN__chromosome_11_screen.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="schraivogel_2020_tap_screen_chr8"><a class="viewcode-back" href="../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr8.html#pertpy.data.schraivogel_2020_tap_screen_chr8">[docs]</a><span class="k">def</span> <span class="nf">schraivogel_2020_tap_screen_chr8</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;TAP-seq applied to K562 cells (only chromosome 8).</span>

<span class="sd">    TAP-seq was applied to generate perturbation-based enhancer–target gene maps in K562 cells.</span>
<span class="sd">    They perturbed all 1,778 putatively active enhancers predicted on the basis of ENCODE data</span>
<span class="sd">    in two regions on chromosome 8 and 11, and identified effects on expressed protein-coding genes</span>
<span class="sd">    within the same regions. Thus, in each cell, 68 (chromosome 8) or 79 (chromosome 11) target genes were measured.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41592-020-0837-5</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;schraivogel_2020_tap_screen_chr8.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SchraivogelSteinmetz2020_TAP_SCREEN__chromosome_8_screen.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="shifrut_2018"><a class="viewcode-back" href="../../../usage/data/pertpy.data.shifrut_2018.html#pertpy.data.shifrut_2018">[docs]</a><span class="k">def</span> <span class="nf">shifrut_2018</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;CD8 T-cells from two donors for two conditions (SLICE and CROP-seq).</span>

<span class="sd">    The authors developed a new method, single guide RNA (sgRNA) lentiviral infection</span>
<span class="sd">    with Cas9 protein electroporation (SLICE), and adapted it to allow for CROP-Seq in</span>
<span class="sd">    primary human T cells. They used a library of 48 sgRNA, derived from GW screens,</span>
<span class="sd">    to explore transcriptional changes downstream of CRISPR-KO. Dataset includes CD8</span>
<span class="sd">    T cells from two donors, for two conditions: with TCR stimulation or No stimulation.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.cell.2018.10.024</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;shifrut_2018.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ShifrutMarson2018.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="srivatsan_2020_sciplex2"><a class="viewcode-back" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex2.html#pertpy.data.srivatsan_2020_sciplex2">[docs]</a><span class="k">def</span> <span class="nf">srivatsan_2020_sciplex2</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;A549 cells exposed to four compounds.</span>

<span class="sd">    A549, a human lung adenocarcinoma cell line, was exposed to one of four compounds:</span>
<span class="sd">    dexamethasone (a corticosteroid agonist), nutlin-3a (a p53-Mdm2 antagonist),</span>
<span class="sd">    BMS-345541 (an inhibitor of nuclear factor κB–dependent transcription), or vorinostat</span>
<span class="sd">    [suberoylanilide hydroxamic acid (SAHA), an HDAC inhibitor], for 24 hours across seven</span>
<span class="sd">    doses in triplicate for a total of 84 drug–dose–replicate combinations and additional</span>
<span class="sd">    vehicle controls. Nuclei from each well were labelled and subjected to sci-RNA-seq.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1126/science.aax6234</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;srivatsan_2020_sciplex2.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SrivatsanTrapnell2020_sciplex2.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="srivatsan_2020_sciplex3"><a class="viewcode-back" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex3.html#pertpy.data.srivatsan_2020_sciplex3">[docs]</a><span class="k">def</span> <span class="nf">srivatsan_2020_sciplex3</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomes of 650000 A549, K562, and mCF7 cells exposed to 188 compounds.</span>

<span class="sd">    sci-Plex was used to screen three well-characterized human cancer cell lines, A549</span>
<span class="sd">    (lung adenocarcinoma), K562 (chronic myelogenous leukemia), and MCF7 (mammary</span>
<span class="sd">    adenocarcinoma)exposed to 188 compounds, profiling ~650,000 single-cell</span>
<span class="sd">    transcriptomes across ~5000 independent samples in one experiment.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1126/science.aax6234</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;srivatsan_2020_sciplex3.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SrivatsanTrapnell2020_sciplex3.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="srivatsan_2020_sciplex4"><a class="viewcode-back" href="../../../usage/data/pertpy.data.srivatsan_2020_sciplex4.html#pertpy.data.srivatsan_2020_sciplex4">[docs]</a><span class="k">def</span> <span class="nf">srivatsan_2020_sciplex4</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;A549 and MCF7 cells treated with pracinostat.</span>

<span class="sd">    A549 and MCF7 cells were treated with pracinostat in the presence and absence of</span>
<span class="sd">    acetyl-CoA precursors (acetate, pyruvate, or citrate) or inhibitors of enzymes</span>
<span class="sd">    (ACLY, ACSS2, or PDH) involved in replenishing acetyl-CoA pools. After treatment,</span>
<span class="sd">    cells were harvested and processed using sci-Plex and trajectories constructed</span>
<span class="sd">    for each cell line. In both A549 and MCF7 cells, acetate, pyruvate, and citrate</span>
<span class="sd">    supplementation was capable of blocking pracinostat-treated cells from reaching</span>
<span class="sd">    the end of the HDAC inhibitor trajectory.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1126/science.aax6234</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;srivatsan_2020_sciplex4.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/SrivatsanTrapnell2020_sciplex4.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="tian_2019_day7neuron"><a class="viewcode-back" href="../../../usage/data/pertpy.data.tian_2019_day7neuron.html#pertpy.data.tian_2019_day7neuron">[docs]</a><span class="k">def</span> <span class="nf">tian_2019_day7neuron</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomes of 20000 day 7 neurons targeted by 58 gRNAs.</span>

<span class="sd">    The authors performed single-cell RNA sequencing of ∼20,000 day 7 neurons via 10x Genomics</span>
<span class="sd">    platform. Transcripts containing sgRNA sequences were further amplified to facilitate</span>
<span class="sd">    sgRNA identity assignment among a pool of 58 sgRNAs (two sgRNAs targeting 27 selected</span>
<span class="sd">    gene and four non-targeting control sgRNAs). Following sequencing, transcriptomes and</span>
<span class="sd">    sgRNA identities were mapped to individual cells. High data quality for neurons was evident</span>
<span class="sd">    from ∼91,000 mean reads per cell, the median number of ∼4,600 genes detected per cell and</span>
<span class="sd">    ∼8,400 cells to which a unique sgRNA could be assigned after quality control.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.neuron.2019.07.014</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;tian_2019_day7neuron.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/TianKampmann2019_day7neuron.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="tian_2019_ipsc"><a class="viewcode-back" href="../../../usage/data/pertpy.data.tian_2019_ipsc.html#pertpy.data.tian_2019_ipsc">[docs]</a><span class="k">def</span> <span class="nf">tian_2019_ipsc</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 20000 iPSCs targeted by 58 sgRNAs.</span>

<span class="sd">    The authors performed single-cell RNA sequencing of ∼20,000 iPSCs via 10x Genomics</span>
<span class="sd">    platform. Transcripts containing sgRNA sequences were further amplified to facilitate</span>
<span class="sd">    sgRNA identity assignment among a pool of 58 sgRNAs (two sgRNAs targeting 27 selected</span>
<span class="sd">    gene and four non-targeting control sgRNAs). Following sequencing, transcriptomes and</span>
<span class="sd">    sgRNA identities were mapped to individual cells. High data quality for iPSCs was evident</span>
<span class="sd">    from ∼84,000 mean reads per cell, the median number of ∼5,000 genes detected per cell and</span>
<span class="sd">    ∼15,000 cells to which a unique sgRNA could be assigned after quality control.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.neuron.2019.07.014</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;tian_2019_iPSC.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/TianKampmann2019_iPSC.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="tian_2021_crispra"><a class="viewcode-back" href="../../../usage/data/pertpy.data.tian_2021_crispra.html#pertpy.data.tian_2021_crispra">[docs]</a><span class="k">def</span> <span class="nf">tian_2021_crispra</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;CROP-seq of 50000 neurons treated with 374 gRNAs (CRISPRa only).</span>

<span class="sd">    For CRISPRa, 100 genes were inclued in the CROP-seq experiments. The CROP-seq</span>
<span class="sd">    libraries included 2 sgRNAs per gene plus 6 non-targeting control sgRNAs for a</span>
<span class="sd">    total of 206 sgRNAs. On Day 10 after CRISPRa neurons were infected by the</span>
<span class="sd">    CROP-seq sgRNA library, approximately 50,000 CRISPRi neurons were loaded into</span>
<span class="sd">    10X chips with about 25,000 input cells per lane.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41593-021-00862-0</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;tian_2021_crispra.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/TianKampmann2021_CRISPRa.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="tian_2021_crispri"><a class="viewcode-back" href="../../../usage/data/pertpy.data.tian_2021_crispri.html#pertpy.data.tian_2021_crispri">[docs]</a><span class="k">def</span> <span class="nf">tian_2021_crispri</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;CROP-seq of 98000 neurons treated with 374 gRNAs (CRISPRi only).</span>

<span class="sd">    For CRISPRi, 184 genes were inclued in the CROP-seq experiments. The CROP-seq</span>
<span class="sd">    libraries included 2 sgRNAs per gene plus 6 non-targeting control sgRNAs for a</span>
<span class="sd">    total of 374 sgRNAs. On Day 10 after CRISPRi neurons were infected by the</span>
<span class="sd">    CROP-seq sgRNA library, approximately 98,000 CRISPRi neurons were loaded into</span>
<span class="sd">    10X chips with about 25,000 input cells per lane.</span>


<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1038/s41593-021-00862-0</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;tian_2021_crispri.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/TianKampmann2021_CRISPRi.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="weinreb_2020"><a class="viewcode-back" href="../../../usage/data/pertpy.data.weinreb_2020.html#pertpy.data.weinreb_2020">[docs]</a><span class="k">def</span> <span class="nf">weinreb_2020</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Mouse embryonic stem cells under different cytokines across time.</span>

<span class="sd">    The authors developed a tool called LARRY (lineage and RNA recovery) and applied</span>
<span class="sd">    it to mouse embryonic stem cells under different cytokine conditions across time.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;weinreb_2020.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/WeinrebKlein2020.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="xie_2017"><a class="viewcode-back" href="../../../usage/data/pertpy.data.xie_2017.html#pertpy.data.xie_2017">[docs]</a><span class="k">def</span> <span class="nf">xie_2017</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Transcriptomics of 51448 cells generated with Mosaic-seq.</span>

<span class="sd">    Mosaic-seq was applied to 71 constituent enhancers from 15 super-enhancers, this</span>
<span class="sd">    analysis of 51,448 sgRNA-induced transcriptomes finds that only a small number of</span>
<span class="sd">    constituents are major effectors of target gene expression.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1016/j.molcel.2017.03.007</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;xie_2017.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/XieHon2017.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="zhao_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.zhao_2021.html#pertpy.data.zhao_2021">[docs]</a><span class="k">def</span> <span class="nf">zhao_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Multiplexed drug perturbation from freshly resected tumors.</span>

<span class="sd">    Source paper:</span>
<span class="sd">        https://doi.org/10.1186/s13073-021-00894-y</span>

<span class="sd">    This study combines multiplexed drug perturbation in acute slice culture from freshly</span>
<span class="sd">    resected tumors with scRNA-seq to profile transcriptome-wide drug responses in</span>
<span class="sd">    individual patients. They applied this approach to drug perturbations on slices derived</span>
<span class="sd">    from six glioblastoma (GBM) resections to identify conserved drug responses and to one</span>
<span class="sd">    additional GBM resection to identify patient-specific responses.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of scPerturb prepared single-cell perturbation data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;zhaoSims2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://zenodo.org/record/7278143/files/ZhaoSims2021.h5ad?download=1&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>
</pre></div>
        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2021, Lukas Heumos, Theislab
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer no-toc">
      
      
      
    </aside>
  </div>
</div><script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/sphinx_highlight.js"></script>
    <script src="../../../_static/scripts/furo.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/design-tabs.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>