<!doctype html>
<html class="no-js" lang="en">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><link rel="index" title="Index" href="../../../genindex.html" /><link rel="search" title="Search" href="../../../search.html" />

    <meta name="generator" content="sphinx-5.1.1, furo 2022.06.21"/>
        <title>pertpy.data._datasets - pertpy</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo.css?digest=40978830699223671f4072448e654b5958f38b89" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sg_gallery.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/override.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/sphinx_gallery.css" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #003262;
  --color-brand-content: #003262per;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  --code-font-size: var(--font-size--small);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../../index.html"><div class="brand">pertpy</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon no-toc" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../../index.html">
  
  
  
</a><form class="sidebar-search-container" method="get" action="../../../search.html" role="search">
  <input class="sidebar-search" placeholder=Search name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../../usage/usage.html">Usage</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.burczynski_crohn.html">pertpy.data.burczynski_crohn</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.papalexi_2021.html">pertpy.data.papalexi_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.kang_2018.html">pertpy.data.kang_2018</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.sc_sim_augur.html">pertpy.data.sc_sim_augur</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.bhattacherjee.html">pertpy.data.bhattacherjee</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.sciplex3_raw.html">pertpy.data.sciplex3_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021.html">pertpy.data.frangieh_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.frangieh_2021_raw.html">pertpy.data.frangieh_2021_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dixit_2016.html">pertpy.data.dixit_2016</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dixit_2016_raw.html">pertpy.data.dixit_2016_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.norman_2019.html">pertpy.data.norman_2019</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.norman_2019_raw.html">pertpy.data.norman_2019_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/data/pertpy.data.dialogue_example.html">pertpy.data.dialogue_example</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Augurpy.html">pertpy.tools.Augurpy</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.Mixscape.html">pertpy.tools.Mixscape</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/tools/pertpy.tools.kernel_pca.html">pertpy.tools.kernel_pca</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.dp_scatter.html">pertpy.plot.ag.dp_scatter</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.important_features.html">pertpy.plot.ag.important_features</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.lollipop.html">pertpy.plot.ag.lollipop</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ag.scatterplot.html">pertpy.plot.ag.scatterplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.violin.html">pertpy.plot.ms.violin</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.perturbscore.html">pertpy.plot.ms.perturbscore</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.heatmap.html">pertpy.plot.ms.heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.barplot.html">pertpy.plot.ms.barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../usage/plot/pertpy.plot.ms.lda.html">pertpy.plot.ms.lda</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">Contributor Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../code_of_conduct.html">Contributor Covenant Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/theislab/pertpy/discussions">Discussions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../authors.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../references.html">References</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon no-toc" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <h1>Source code for pertpy.data._datasets</h1><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="kn">import</span> <span class="nn">muon</span> <span class="k">as</span> <span class="nn">mu</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">from</span> <span class="nn">anndata</span> <span class="kn">import</span> <span class="n">AnnData</span>
<span class="kn">from</span> <span class="nn">mudata</span> <span class="kn">import</span> <span class="n">MuData</span>
<span class="kn">from</span> <span class="nn">scanpy</span> <span class="kn">import</span> <span class="n">settings</span>

<span class="kn">from</span> <span class="nn">pertpy.data._dataloader</span> <span class="kn">import</span> <span class="n">_download</span>


<div class="viewcode-block" id="burczynski_crohn"><a class="viewcode-back" href="../../../usage/data/pertpy.data.burczynski_crohn.html#pertpy.data.burczynski_crohn">[docs]</a><span class="k">def</span> <span class="nf">burczynski_crohn</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Bulk data with conditions ulcerative colitis (UC) and Crohn&#39;s disease (CD).</span>

<span class="sd">    The study assesses transcriptional profiles in peripheral blood mononuclear</span>
<span class="sd">    cells from 42 healthy individuals, 59 CD patients, and 26 UC patients by</span>
<span class="sd">    hybridization to microarrays interrogating more than 22,000 sequences.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Burczynski et al., &quot;Molecular classification of Crohn&#39;s disease and</span>
<span class="sd">        ulcerative colitis patients using transcriptional profiles in peripheral blood mononuclear cells&quot;</span>
<span class="sd">        J Mol Diagn 8, 51 (2006). PMID:16436634.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of the Burczynski dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">return</span> <span class="n">sc</span><span class="o">.</span><span class="n">datasets</span><span class="o">.</span><span class="n">burczynski06</span><span class="p">()</span></div>


<div class="viewcode-block" id="papalexi_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.papalexi_2021.html#pertpy.data.papalexi_2021">[docs]</a><span class="k">def</span> <span class="nf">papalexi_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">MuData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Dataset of the Mixscape Vignette.</span>

<span class="sd">    https://satijalab.org/seurat/articles/mixscape_vignette.html</span>

<span class="sd">    Reference:</span>
<span class="sd">    Papalexi, E., Mimitou, E.P., Butler, A.W. et al. Characterizing the molecular regulation</span>
<span class="sd">    of inhibitory immune checkpoints with multimodal single-cell screens.</span>
<span class="sd">    Nat Genet 53, 322–331 (2021). https://doi.org/10.1038/s41588-021-00778-2</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of the Crispr dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;papalexi_2021.h5mu&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/36509460&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">mudata</span> <span class="o">=</span> <span class="n">mu</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">mudata</span> <span class="o">=</span> <span class="n">mu</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">mudata</span></div>


<div class="viewcode-block" id="sc_sim_augur"><a class="viewcode-back" href="../../../usage/data/pertpy.data.sc_sim_augur.html#pertpy.data.sc_sim_augur">[docs]</a><span class="k">def</span> <span class="nf">sc_sim_augur</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Simulated test dataset used the Usage example for the Augur.</span>

<span class="sd">    https://github.com/neurorestore/Augur</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a simulated single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;sc_sim_augur.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/31645886&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="bhattacherjee"><a class="viewcode-back" href="../../../usage/data/pertpy.data.bhattacherjee.html#pertpy.data.bhattacherjee">[docs]</a><span class="k">def</span> <span class="nf">bhattacherjee</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Processed single-cell data PFC adult mice under cocaine self-administration.</span>

<span class="sd">    Adult mice were subject to cocaine self-administration, samples were</span>
<span class="sd">    collected a three time points: Maintance, 48h after cocaine withdrawal and</span>
<span class="sd">    15 days after cocaine withdrawal.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Bhattacherjee A, Djekidel MN, Chen R, Chen W, Tuesta LM, Zhang Y. Cell</span>
<span class="sd">        type-specific transcriptional programs in mouse prefrontal cortex during</span>
<span class="sd">        adolescence and addiction. Nat Commun. 2019 Sep 13;10(1):4169.</span>
<span class="sd">        doi: 10.1038/s41467-019-12054-3. PMID: 31519873; PMCID: PMC6744514.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;bhattacherjee_rna.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34526528&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="sciplex3_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.sciplex3_raw.html#pertpy.data.sciplex3_raw">[docs]</a><span class="k">def</span> <span class="nf">sciplex3_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Raw sciplex3 perturbation dataset curated for perturbation modeling.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Srivatsan SR, McFaline-Figueroa JL, Ramani V, Saunders L, Cao J, Packer J,</span>
<span class="sd">        Pliner HA, Jackson DL, Daza RM, Christiansen L, Zhang F, Steemers F,</span>
<span class="sd">        Shendure J, Trapnell C. Massively multiplex chemical transcriptomics at</span>
<span class="sd">        single-cell resolution. Science. 2020 Jan 3;367(6473):45-51.</span>
<span class="sd">        doi: 10.1126/science.aax6234. Epub 2019 Dec 5. PMID: 31806696; PMCID: PMC7289078.</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a single-cell RNA seq dataset</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;sciplex3.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/33979517&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021.html#pertpy.data.frangieh_2021">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Processed perturb-CITE-seq data with multi-modal RNA and protein single-cell profiling.</span>

<span class="sd">    We profiled RNA and 20 surface proteins in over 218,000 cells under ~750 perturbations,</span>
<span class="sd">    chosen by their membership in an immune evasion program that is associated with</span>
<span class="sd">    immunotherapy resistance in patients.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Frangieh, C.J., Melms, J.C., Thakore, P.I. et al. Multimodal pooled Perturb-CITE-seq</span>
<span class="sd">        screens in patient models define mechanisms of cancer immune evasion.</span>
<span class="sd">        Nat Genet 53, 332–341 (2021). https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of a Perturb-CITE-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34013717&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="frangieh_2021_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.frangieh_2021_raw.html#pertpy.data.frangieh_2021_raw">[docs]</a><span class="k">def</span> <span class="nf">frangieh_2021_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Raw Perturb-CITE-seq data with multi-modal RNA and protein single-cell profiling readout.</span>

<span class="sd">    We profiled RNA and 20 surface proteins in over 218,000 cells under ~750 perturbations,</span>
<span class="sd">    chosen by their membership in an immune evasion program that is associated with</span>
<span class="sd">    immunotherapy resistance in patients.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Frangieh, C.J., Melms, J.C., Thakore, P.I. et al. Multimodal pooled Perturb-CITE-seq</span>
<span class="sd">        screens in patient models define mechanisms of cancer immune evasion.</span>
<span class="sd">        Nat Genet 53, 332–341 (2021). https://doi.org/10.1038/s41588-021-00779-1</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of raw Perturb-CITE-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;frangieh_2021.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34012565&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dixit_2016_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dixit_2016_raw.html#pertpy.data.dixit_2016_raw">[docs]</a><span class="k">def</span> <span class="nf">dixit_2016_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Perturb-seq: scRNA-seq with pooled CRISPR-KO perturbations.</span>

<span class="sd">    scRNA-seq with pooled CRISPR-KO perturbations in 200,000 cells across six screens</span>
<span class="sd">    unstimulated BMDC, BMDC stimulated at 3hr, TFs in K562 at 7 and 13 days post transduction,</span>
<span class="sd">    and 13 days at a higher MOI of perturbations.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Dixit A, Parnas O, Li B, Chen J et al. Perturb-Seq: Dissecting Molecular Circuits with</span>
<span class="sd">        Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.</span>
<span class="sd">        Cell 2016 Dec 15;167(7):1853-1866.e17. DOI:https://doi.org/10.1016/j.cell.2016.11.038</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of raw Perturb-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dixit_2016_raw.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34012565&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dixit_2016"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dixit_2016.html#pertpy.data.dixit_2016">[docs]</a><span class="k">def</span> <span class="nf">dixit_2016</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Perturb-seq: scRNA-seq with pooled CRISPR-KO perturbations.</span>

<span class="sd">    scRNA-seq with pooled CRISPR-KO perturbations in 200,000 cells across six screens</span>
<span class="sd">    unstimulated BMDC, BMDC stimulated at 3hr, TFs in K562 at 7 and 13 days post trasnduction,</span>
<span class="sd">    and 13 days at a higher MOI of perturbations.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Dixit A, Parnas O, Li B, Chen J et al. Perturb-Seq: Dissecting Molecular Circuits with</span>
<span class="sd">        Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.</span>
<span class="sd">        Cell 2016 Dec 15;167(7):1853-1866.e17. DOI:https://doi.org/10.1016/j.cell.2016.11.038</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of Perturb-seq data</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dixit_2016.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34014608&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="norman_2019"><a class="viewcode-back" href="../../../usage/data/pertpy.data.norman_2019.html#pertpy.data.norman_2019">[docs]</a><span class="k">def</span> <span class="nf">norman_2019</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Processed single-cell, pooled CRISPR screening.</span>

<span class="sd">    Single-cell, pooled CRISPR screening experiment comparing the transcriptional effects of</span>
<span class="sd">    overexpressing genes alone or in combination</span>

<span class="sd">    Reference:</span>
<span class="sd">        Norman, Thomas M et al. “Exploring genetic interaction manifolds constructed from rich</span>
<span class="sd">        single-cell phenotypes.” Science (New York, N.Y.) vol. 365,6455 (2019): 786-793.</span>
<span class="sd">        doi:10.1126/science.aax4438</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of sc pooled CRISPR screening</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;norman_2019.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34027562&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="norman_2019_raw"><a class="viewcode-back" href="../../../usage/data/pertpy.data.norman_2019_raw.html#pertpy.data.norman_2019_raw">[docs]</a><span class="k">def</span> <span class="nf">norman_2019_raw</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Raw single-cell, pooled CRISPR screening.</span>

<span class="sd">    Single-cell, pooled CRISPR screening experiment comparing the transcriptional effects of</span>
<span class="sd">    overexpressing genes alone or in combination</span>

<span class="sd">    Reference:</span>
<span class="sd">        Norman, Thomas M et al. “Exploring genetic interaction manifolds constructed from rich</span>
<span class="sd">        single-cell phenotypes.” Science (New York, N.Y.) vol. 365,6455 (2019): 786-793.</span>
<span class="sd">        doi:10.1126/science.aax4438</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of sc pooled CRISPR screening</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;norman_2019_raw.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34002548&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="dialogue_example"><a class="viewcode-back" href="../../../usage/data/pertpy.data.dialogue_example.html#pertpy.data.dialogue_example">[docs]</a><span class="k">def</span> <span class="nf">dialogue_example</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Example dataset used to feature DIALOGUE.</span>

<span class="sd">    https://github.com/livnatje/DIALOGUE/wiki/Example</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;dialogue_example.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34490714&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>


<div class="viewcode-block" id="kang_2018"><a class="viewcode-back" href="../../../usage/data/pertpy.data.kang_2018.html#pertpy.data.kang_2018">[docs]</a><span class="k">def</span> <span class="nf">kang_2018</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>  <span class="c1"># pragma: no cover</span>
    <span class="sd">&quot;&quot;&quot;Processed multiplexing droplet-based single cell RNA-sequencing using genetic barcodes</span>

<span class="sd">    HiSeq 2500 data for sequencing of PBMCs from SLE patients and 2 controls. We collected 1M cells</span>
<span class="sd">    each from frozen PBMC samples that were Ficoll isolated and prepared using the 10x Single Cell</span>
<span class="sd">    instrument according to standard protocol. Samples A, B, and C were prepared on the instrument</span>
<span class="sd">    directly following thaw, while samples 2.1 and 2.2 were cultured for 6 hours with (B) or</span>
<span class="sd">    without (A) IFN-beta stimulation prior to loading on the 10x Single Cell instrument.</span>

<span class="sd">    Reference:</span>
<span class="sd">        Kang, H., Subramaniam, M., Targ, S. et al. Multiplexed droplet single-cell RNA-sequencing</span>
<span class="sd">        using natural genetic variation.</span>
<span class="sd">        Nat Biotechnol 36, 89–94 (2018). https://doi.org/10.1038/nbt.4042</span>

<span class="sd">    Returns:</span>
<span class="sd">        :class:`~anndata.AnnData` object of droplet-based single cell RNA-sequencing</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">output_file_name</span> <span class="o">=</span> <span class="s2">&quot;kang_2018.h5ad&quot;</span>
    <span class="n">output_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
        <span class="n">_download</span><span class="p">(</span>
            <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34464122&quot;</span><span class="p">,</span>
            <span class="n">output_file_name</span><span class="o">=</span><span class="n">output_file_name</span><span class="p">,</span>
            <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="p">,</span>
            <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">datasetdir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">output_file_name</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="n">output_file_path</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">adata</span></div>
</pre></div>
        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2021, Lukas Heumos, Theislab
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer no-toc">
      
      
      
    </aside>
  </div>
</div><script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/scripts/furo.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/design-tabs.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>