<!doctype html>
<html class="no-js" lang="en">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><link rel="index" title="Index" href="../../../../genindex.html" /><link rel="search" title="Search" href="../../../../search.html" />

    <!-- Generated with Sphinx 7.1.2 and Furo 2023.07.26 -->
        <title>pertpy.tools._metadata._cell_line - pertpy</title>
      <link rel="stylesheet" type="text/css" href="../../../../_static/pygments.css?v=80d5e7a1" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/styles/furo.css?v=135e06be" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/sg_gallery.css?v=61a4c737" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/design-style.1e8bd061cd6da7fc9cf755528e8ffc24.min.css?v=0a3b3ea7" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/styles/furo-extensions.css?v=36a5483c" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/css/override.css" />
    <link rel="stylesheet" type="text/css" href="../../../../_static/css/sphinx_gallery.css?v=4d525498" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --color-brand-primary: #003262;
  --color-brand-content: #003262per;
  --admonition-font-size: var(--font-size-normal);
  --admonition-title-font-size: var(--font-size-normal);
  --code-font-size: var(--font-size--small);
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../../../index.html"><div class="brand">pertpy</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon no-toc" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../../../index.html">
  
  
  
</a><form class="sidebar-search-container" method="get" action="../../../../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../installation.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../../../usage/usage.html">Usage</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of Usage</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.adamson_2016_pilot.html">pertpy.data.adamson_2016_pilot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.adamson_2016_upr_epistasis.html">pertpy.data.adamson_2016_upr_epistasis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.adamson_2016_upr_perturb_seq.html">pertpy.data.adamson_2016_upr_perturb_seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.aissa_2021.html">pertpy.data.aissa_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.bhattacherjee.html">pertpy.data.bhattacherjee</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.burczynski_crohn.html">pertpy.data.burczynski_crohn</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.chang_2021.html">pertpy.data.chang_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.datlinger_2017.html">pertpy.data.datlinger_2017</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.datlinger_2021.html">pertpy.data.datlinger_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.dialogue_example.html">pertpy.data.dialogue_example</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.dixit_2016.html">pertpy.data.dixit_2016</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.dixit_2016_raw.html">pertpy.data.dixit_2016_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.dixit_2016_scperturb.html">pertpy.data.dixit_2016_scperturb</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.frangieh_2021.html">pertpy.data.frangieh_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.frangieh_2021_protein.html">pertpy.data.frangieh_2021_protein</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.frangieh_2021_raw.html">pertpy.data.frangieh_2021_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.frangieh_2021_rna.html">pertpy.data.frangieh_2021_rna</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.gasperini_2019_atscale.html">pertpy.data.gasperini_2019_atscale</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.gasperini_2019_highmoi.html">pertpy.data.gasperini_2019_highmoi</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.gasperini_2019_lowmoi.html">pertpy.data.gasperini_2019_lowmoi</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.gehring_2019.html">pertpy.data.gehring_2019</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.haber_2017_regions.html">pertpy.data.haber_2017_regions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.kang_2018.html">pertpy.data.kang_2018</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.mcfarland_2020.html">pertpy.data.mcfarland_2020</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.norman_2019.html">pertpy.data.norman_2019</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.norman_2019_raw.html">pertpy.data.norman_2019_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.papalexi_2021.html">pertpy.data.papalexi_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.replogle_2022_k562_essential.html">pertpy.data.replogle_2022_k562_essential</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.replogle_2022_k562_gwps.html">pertpy.data.replogle_2022_k562_gwps</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.replogle_2022_rpe1.html">pertpy.data.replogle_2022_rpe1</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.sc_sim_augur.html">pertpy.data.sc_sim_augur</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.schiebinger_2019_16day.html">pertpy.data.schiebinger_2019_16day</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.schiebinger_2019_18day.html">pertpy.data.schiebinger_2019_18day</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr8.html">pertpy.data.schraivogel_2020_tap_screen_chr8</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.schraivogel_2020_tap_screen_chr11.html">pertpy.data.schraivogel_2020_tap_screen_chr11</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.sciplex3_raw.html">pertpy.data.sciplex3_raw</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.shifrut_2018.html">pertpy.data.shifrut_2018</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.smillie.html">pertpy.data.smillie</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.srivatsan_2020_sciplex2.html">pertpy.data.srivatsan_2020_sciplex2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.srivatsan_2020_sciplex3.html">pertpy.data.srivatsan_2020_sciplex3</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.srivatsan_2020_sciplex4.html">pertpy.data.srivatsan_2020_sciplex4</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.stephenson_2021_subsampled.html">pertpy.data.stephenson_2021_subsampled</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.tian_2019_day7neuron.html">pertpy.data.tian_2019_day7neuron</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.tian_2019_ipsc.html">pertpy.data.tian_2019_ipsc</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.tian_2021_crispra.html">pertpy.data.tian_2021_crispra</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.tian_2021_crispri.html">pertpy.data.tian_2021_crispri</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.weinreb_2020.html">pertpy.data.weinreb_2020</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.xie_2017.html">pertpy.data.xie_2017</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/data/pertpy.data.zhao_2021.html">pertpy.data.zhao_2021</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/preprocessing/pertpy.preprocessing.GuideAssignment.html">pertpy.preprocessing.GuideAssignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Mixscape.html">pertpy.tools.Mixscape</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Milo.html">pertpy.tools.Milo</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Sccoda.html">pertpy.tools.Sccoda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Tasccoda.html">pertpy.tools.Tasccoda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Dialogue.html">pertpy.tools.Dialogue</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Distance.html">pertpy.tools.Distance</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.DistanceTest.html">pertpy.tools.DistanceTest</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html">pertpy.tools.CellLineMetaData</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.Augur.html">pertpy.tools.Augur</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.SCGEN.html">pertpy.tools.SCGEN</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.DiscriminatorClassifierSpace.html">pertpy.tools.DiscriminatorClassifierSpace</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.CentroidSpace.html">pertpy.tools.CentroidSpace</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.DBSCANSpace.html">pertpy.tools.DBSCANSpace</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.KMeansSpace.html">pertpy.tools.KMeansSpace</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/tools/pertpy.tools.PseudobulkSpace.html">pertpy.tools.PseudobulkSpace</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.guide.html">pertpy.plot.guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ms.violin.html">pertpy.plot.ms.violin</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ms.perturbscore.html">pertpy.plot.ms.perturbscore</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ms.heatmap.html">pertpy.plot.ms.heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ms.barplot.html">pertpy.plot.ms.barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ms.lda.html">pertpy.plot.ms.lda</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.milo.nhood_graph.html">pertpy.plot.milo.nhood_graph</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.milo.nhood.html">pertpy.plot.milo.nhood</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.milo.da_beeswarm.html">pertpy.plot.milo.da_beeswarm</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.milo.nhood_counts_by_cond.html">pertpy.plot.milo.nhood_counts_by_cond</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.stacked_barplot.html">pertpy.plot.coda.stacked_barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.effects_barplot.html">pertpy.plot.coda.effects_barplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.boxplots.html">pertpy.plot.coda.boxplots</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.rel_abundance_dispersion_plot.html">pertpy.plot.coda.rel_abundance_dispersion_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.draw_tree.html">pertpy.plot.coda.draw_tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.draw_effects.html">pertpy.plot.coda.draw_effects</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.coda.effects_umap.html">pertpy.plot.coda.effects_umap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ag.dp_scatter.html">pertpy.plot.ag.dp_scatter</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ag.important_features.html">pertpy.plot.ag.important_features</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ag.lollipop.html">pertpy.plot.ag.lollipop</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.ag.scatterplot.html">pertpy.plot.ag.scatterplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.scg.reg_mean_plot.html">pertpy.plot.scg.reg_mean_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.scg.reg_var_plot.html">pertpy.plot.scg.reg_var_plot</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../usage/plot/pertpy.plot.scg.binary_classifier.html">pertpy.plot.scg.binary_classifier</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../contributing.html">Contributor Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../code_of_conduct.html">Contributor Covenant Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/theislab/pertpy/discussions">Discussions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../authors.html">Credits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../references.html">References</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon no-toc" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <h1>Source code for pertpy.tools._metadata._cell_line</h1><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">annotations</span>

<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>

<span class="kn">import</span> <span class="nn">anndata</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">anndata</span> <span class="kn">import</span> <span class="n">AnnData</span>
<span class="kn">from</span> <span class="nn">rich</span> <span class="kn">import</span> <span class="nb">print</span>
<span class="kn">from</span> <span class="nn">scanpy</span> <span class="kn">import</span> <span class="n">settings</span>

<span class="kn">from</span> <span class="nn">pertpy.data._dataloader</span> <span class="kn">import</span> <span class="n">_download</span>


<div class="viewcode-block" id="CellLineMetaData"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData">[docs]</a><span class="k">class</span> <span class="nc">CellLineMetaData</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;Utilities to fetch cell line metadata.&quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="c1"># Download cell line metadata from DepMap</span>
        <span class="n">cell_line_file_name</span> <span class="o">=</span> <span class="s2">&quot;sample_info.csv&quot;</span>
        <span class="n">cell_line_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">cell_line_file_name</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">cell_line_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;[bold yellow]No DepMap metadata file found. Starting download now.&quot;</span><span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://ndownloader.figshare.com/files/35020903&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">cell_line_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cell_line_meta</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">cell_line_file_path</span><span class="p">)</span>

        <span class="c1"># Download cell line metadata from The Genomics of Drug Sensitivity in Cancer Project</span>
        <span class="n">cell_line_cancer_project_file_name</span> <span class="o">=</span> <span class="s2">&quot;cell_line_cancer_project.csv&quot;</span>
        <span class="n">cell_line_cancer_project_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">cell_line_cancer_project_file_name</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">cell_line_cancer_project_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]No cell line metadata file from The Genomics of Drug Sensitivity &quot;</span>
                <span class="s2">&quot;in Cancer Project found. Starting download now.&quot;</span>
            <span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://www.cancerrxgene.org/api/celllines?list=all&amp;sEcho=1&amp;iColumns=7&amp;sColumns=&amp;&quot;</span>
                <span class="s2">&quot;iDisplayStart=0&amp;iDisplayLength=25&amp;mDataProp_0=0&amp;mDataProp_1=1&amp;mDataProp_2=2&amp;mDataProp_3=3&amp;&quot;</span>
                <span class="s2">&quot;mDataProp_4=4&amp;mDataProp_5=5&amp;mDataProp_6=6&amp;sSearch=&amp;bRegex=false&amp;sSearch_0=&amp;bRegex_0=false&amp;&quot;</span>
                <span class="s2">&quot;bSearchable_0=true&amp;sSearch_1=&amp;bRegex_1=false&amp;bSearchable_1=true&amp;sSearch_2=&amp;bRegex_2=false&amp;&quot;</span>
                <span class="s2">&quot;bSearchable_2=true&amp;sSearch_3=&amp;bRegex_3=false&amp;bSearchable_3=true&amp;sSearch_4=&amp;bRegex_4=false&amp;&quot;</span>
                <span class="s2">&quot;bSearchable_4=true&amp;sSearch_5=&amp;bRegex_5=false&amp;bSearchable_5=true&amp;sSearch_6=&amp;bRegex_6=false&amp;&quot;</span>
                <span class="s2">&quot;bSearchable_6=true&amp;iSortCol_0=0&amp;sSortDir_0=asc&amp;iSortingCols=1&amp;bSortable_0=true&amp;bSortable_1=true&amp;&quot;</span>
                <span class="s2">&quot;bSortable_2=true&amp;bSortable_3=true&amp;bSortable_4=true&amp;bSortable_5=true&amp;bSortable_6=true&amp;export=csv&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">cell_line_cancer_project_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
            <span class="p">)</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">cell_line_cancer_project_file_path</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="p">[</span><span class="s2">&quot;stripped_cell_line_name&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="p">[</span>
            <span class="s2">&quot;Cell line Name&quot;</span>
        <span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;\-|\.&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">regex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="p">[</span><span class="s2">&quot;stripped_cell_line_name&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Categorical</span><span class="p">(</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="p">[</span><span class="s2">&quot;stripped_cell_line_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
        <span class="p">)</span>
        <span class="c1"># pivot the data frame so that each cell line has only one row of metadata</span>
        <span class="n">index_col</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span> <span class="o">-</span> <span class="p">{</span><span class="s2">&quot;Datasets&quot;</span><span class="p">,</span> <span class="s2">&quot;number of drugs&quot;</span><span class="p">})</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">pivot</span><span class="p">(</span>
            <span class="n">index</span><span class="o">=</span><span class="n">index_col</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="s2">&quot;Datasets&quot;</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="s2">&quot;number of drugs&quot;</span>
        <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

        <span class="c1"># Download metadata for driver genes of the intOGen analysis from DepMap_Sanger</span>
        <span class="n">driver_gene_intOGen_file_name</span> <span class="o">=</span> <span class="s2">&quot;IntOGen-Drivers.zip&quot;</span>
        <span class="n">driver_gene_intOGen_file_path</span> <span class="o">=</span> <span class="p">(</span>
            <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/2020-02-02_IntOGen-Drivers-20200213/Compendium_Cancer_Genes.tsv&quot;</span>
        <span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">driver_gene_intOGen_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]No metadata file was found for driver genes of the intOGen analysis.&quot;</span>
                <span class="s2">&quot; Starting download now.&quot;</span>
            <span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://www.intogen.org/download?file=IntOGen-Drivers-20200201.zip&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">driver_gene_intOGen_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_intOGen</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="n">driver_gene_intOGen_file_path</span><span class="p">)</span>

        <span class="c1"># Download metadata for driver genes of the COSMIC Tier 1 gene</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_cosmic</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;https://www.dropbox.com/s/8azkmt7vqz56e2m/COSMIC_tier1.csv?dl=1&quot;</span><span class="p">)</span>

        <span class="c1"># Download bulk RNA-seq data collated from the Wellcome Sanger Institute and the Broad Institute</span>
        <span class="n">bulk_rna_sanger_file_name</span> <span class="o">=</span> <span class="s2">&quot;rnaseq_sanger.zip&quot;</span>
        <span class="n">bulk_rna_sanger_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/rnaseq_sanger_20210316.csv&quot;</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">bulk_rna_sanger_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]No metadata file was found for bulk RNA-seq data of Sanger cell line.&quot;</span>
                <span class="s2">&quot; Starting download now.&quot;</span>
            <span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://cog.sanger.ac.uk/cmp/download/rnaseq_sanger_20210316.zip&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">bulk_rna_sanger_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_sanger</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">bulk_rna_sanger_file_path</span><span class="p">)</span>

        <span class="n">bulk_rna_broad_file_name</span> <span class="o">=</span> <span class="s2">&quot;rnaseq_broad.zip&quot;</span>
        <span class="n">bulk_rna_broad_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/rnaseq_broad_20210317.csv&quot;</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">bulk_rna_broad_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]No metadata file was found for bulk RNA-seq data of broad cell line.&quot;</span>
                <span class="s2">&quot; Starting download now.&quot;</span>
            <span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://cog.sanger.ac.uk/cmp/download/rnaseq_broad_20210317.zip&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">bulk_rna_broad_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_broad</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">bulk_rna_broad_file_path</span><span class="p">)</span>

        <span class="c1"># Download proteomics data from ProCan-DepMapSanger</span>
        <span class="n">proteomics_file_name</span> <span class="o">=</span> <span class="s2">&quot;Proteomics_20221214.zip&quot;</span>
        <span class="n">proteomics_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/proteomics_all_20221214.csv&quot;</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">proteomics_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]No metadata file was found for proteomics data (ProCan-DepMapSanger).&quot;</span>
                <span class="s2">&quot; Starting download now.&quot;</span>
            <span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://cog.sanger.ac.uk/cmp/download/Proteomics_20221214.zip&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">proteomics_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">proteomics_file_path</span><span class="p">)</span>

        <span class="c1"># Download CCLE expression data from DepMap</span>
        <span class="n">ccle_expr_file_name</span> <span class="o">=</span> <span class="s2">&quot;CCLE_expression.csv&quot;</span>
        <span class="n">ccle_expr_file_path</span> <span class="o">=</span> <span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="o">.</span><span class="fm">__str__</span><span class="p">()</span> <span class="o">+</span> <span class="s2">&quot;/CCLE_expression.csv&quot;</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">Path</span><span class="p">(</span><span class="n">ccle_expr_file_path</span><span class="p">)</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;[bold yellow]No metadata file was found for CCLE expression data.&quot;</span> <span class="s2">&quot; Starting download now.&quot;</span><span class="p">)</span>
            <span class="n">_download</span><span class="p">(</span>
                <span class="n">url</span><span class="o">=</span><span class="s2">&quot;https://figshare.com/ndownloader/files/34989919&quot;</span><span class="p">,</span>
                <span class="n">output_file_name</span><span class="o">=</span><span class="n">ccle_expr_file_name</span><span class="p">,</span>
                <span class="n">output_path</span><span class="o">=</span><span class="n">settings</span><span class="o">.</span><span class="n">cachedir</span><span class="p">,</span>
                <span class="n">is_zip</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ccle_expr</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">ccle_expr_file_path</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<div class="viewcode-block" id="CellLineMetaData.getinfo_annotate_driver_genes"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.getinfo_annotate_driver_genes">[docs]</a>    <span class="k">def</span> <span class="nf">getinfo_annotate_driver_genes</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">driver_gene_set</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;intOGen&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A brief summary of genes in cancer driver annotation data.</span>

<span class="sd">        Args:</span>
<span class="sd">            driver_gene_set: gene set for cancer driver annotation: intOGen or COSMIC. Defaults to &quot;intOGen&quot;.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">driver_gene_set</span> <span class="o">==</span> <span class="s2">&quot;intOGen&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the DepMap_Sanger driver gene annotation for intOGen genes: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_intOGen</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_intOGen</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="si">}</span><span class="s2"> driver genes are saved in this file&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">driver_gene_set</span> <span class="o">==</span> <span class="s2">&quot;COSMIC&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the DepMap_Sanger driver gene annotation for COSMIC Tier 1 genes: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_cosmic</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">driver_gene_cosmic</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="si">}</span><span class="s2"> driver genes are saved in this file&quot;</span><span class="p">)</span></div>

<div class="viewcode-block" id="CellLineMetaData.getinfo_annotate_bulk_rna_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.getinfo_annotate_bulk_rna_expression">[docs]</a>    <span class="k">def</span> <span class="nf">getinfo_annotate_bulk_rna_expression</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cell_line_source</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;broad&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A brief summary of bulk RNA expression data.</span>

<span class="sd">        Args:</span>
<span class="sd">            cell_line_source: the source of RNA-seq data: broad or sanger. Defaults to&quot;broad&quot;.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">cell_line_source</span> <span class="o">==</span> <span class="s2">&quot;broad&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the RNA-Seq Data for broad cell line only: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_broad</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_broad</span><span class="o">.</span><span class="n">model_name</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique cell lines are saved in this file.&quot;</span><span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_broad</span><span class="o">.</span><span class="n">gene_id</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique genes are saved in this file&quot;</span><span class="p">)</span>

        <span class="k">elif</span> <span class="n">cell_line_source</span> <span class="o">==</span> <span class="s2">&quot;sanger&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the RNA-Seq Data for Sanger cell line only: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_sanger</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_sanger</span><span class="o">.</span><span class="n">model_name</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique cell lines are saved in this file.&quot;</span><span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_sanger</span><span class="o">.</span><span class="n">gene_id</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique genes are saved in this file.&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified source of bulk rna expression data is not available. &quot;</span>
                <span class="s2">&quot;Please choose either broad or sanger.&quot;</span>
            <span class="p">)</span></div>

<div class="viewcode-block" id="CellLineMetaData.getinfo_annotate_cell_lines"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.getinfo_annotate_cell_lines">[docs]</a>    <span class="k">def</span> <span class="nf">getinfo_annotate_cell_lines</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cell_line_source</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;DepMap&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A brief summary of cell line metadata.</span>

<span class="sd">        Args:</span>
<span class="sd">            cell_line_source: the source of cell line annotation: DepMap or Cancerrxgene. Defaults to &quot;DepMap&quot;.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">cell_line_source</span> <span class="o">==</span> <span class="s2">&quot;DepMap&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the DepMap cell line annotation: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cell_line_meta</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cell_line_meta</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="si">}</span><span class="s2"> cell lines are saved in this file&quot;</span><span class="p">)</span>

        <span class="k">elif</span> <span class="n">cell_line_source</span> <span class="o">==</span> <span class="s2">&quot;Cancerrxgene&quot;</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;Current available information in the cell line annotation from the project &quot;</span>
                <span class="s2">&quot;Genomics of Drug Sensitivity in Cancer: &quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="si">}</span><span class="s2"> cell lines are saved in this file&quot;</span><span class="p">)</span>

        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified source of cell line annotation is not available. &quot;</span>
                <span class="s2">&quot;Please choose either DepMap or Cancerrxgene.&quot;</span>
            <span class="p">)</span></div>

<div class="viewcode-block" id="CellLineMetaData.getinfo_annotate_protein_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.getinfo_annotate_protein_expression">[docs]</a>    <span class="k">def</span> <span class="nf">getinfo_annotate_protein_expression</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A brief summary of protein expression data.&quot;&quot;&quot;</span>
        <span class="nb">print</span><span class="p">(</span>
            <span class="s2">&quot;Current available information in the proteomics data: &quot;</span><span class="p">,</span>
            <span class="o">*</span><span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">),</span>
            <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="o">.</span><span class="n">model_name</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique cell lines are saved in this file.&quot;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="o">.</span><span class="n">uniprot_id</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique proteins are saved in this file.&quot;</span><span class="p">)</span></div>

<div class="viewcode-block" id="CellLineMetaData.getinfo_annotate_ccle_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.getinfo_annotate_ccle_expression">[docs]</a>    <span class="k">def</span> <span class="nf">getinfo_annotate_ccle_expression</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A brief summary of CCLE expression data.&quot;&quot;&quot;</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot; Expression of </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ccle_expr</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> genes is saved in this file.&quot;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ccle_expr</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> unique cell lines are available in this file.&quot;</span><span class="p">)</span></div>

<div class="viewcode-block" id="CellLineMetaData.annotate_cell_lines"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.annotate_cell_lines">[docs]</a>    <span class="k">def</span> <span class="nf">annotate_cell_lines</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">adata</span><span class="p">:</span> <span class="n">AnnData</span><span class="p">,</span>
        <span class="n">cell_line_identifier</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;DepMap_ID&quot;</span><span class="p">,</span>
        <span class="n">identifier_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;DepMap_ID&quot;</span><span class="p">,</span>
        <span class="n">cell_line_information</span><span class="p">:</span> <span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
        <span class="n">source</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;DepMap&quot;</span><span class="p">,</span>
        <span class="n">copy</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Fetch cell line annotation.</span>

<span class="sd">        For each cell, we fetch cell line annotation from Dependency Map (DepMap).</span>

<span class="sd">        Args:</span>
<span class="sd">            adata: The AnnData object to annotate.</span>
<span class="sd">            cell_line_identifier: The column of `.obs` with cell line information. Defaults to &quot;DepMap_ID&quot;.</span>
<span class="sd">            identifier_type: The type of cell line information, e.g. DepMap_ID, cell_line_name or</span>
<span class="sd">                stripped_cell_line_name. To fetch cell line metadata from Cancerrxgene, it is recommended to choose</span>
<span class="sd">                &quot;stripped_cell_line_name&quot;. Defaults to &quot;DepMap_ID&quot;.</span>
<span class="sd">            cell_line_information: The metadata to fetch. All metadata will be fetched by default. Defaults to None (=all).</span>
<span class="sd">            source: The source of cell line metadata, DepMap or Cancerrxgene. Defaults to &quot;DepMap&quot;.</span>
<span class="sd">            copy: Determines whether a copy of the AnnData object is returned. Defaults to False.</span>

<span class="sd">        Returns:</span>
<span class="sd">            Returns an AnnData object with cell line annotation.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">copy</span><span class="p">:</span>
            <span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>

        <span class="k">if</span> <span class="n">source</span> <span class="o">==</span> <span class="s2">&quot;DepMap&quot;</span><span class="p">:</span>
            <span class="n">cell_line_meta</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cell_line_meta</span>
        <span class="k">elif</span> <span class="n">source</span> <span class="o">==</span> <span class="s2">&quot;Cancerrxgene&quot;</span><span class="p">:</span>
            <span class="n">cell_line_meta</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">cl_cancer_project_meta</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified source of cell line metadata is not available.&quot;</span>
                <span class="s2">&quot;Please choose either DepMap or Cancerrxgene. Default is DepMap.&quot;</span>
            <span class="p">)</span>

        <span class="k">if</span> <span class="n">identifier_type</span> <span class="ow">in</span> <span class="n">cell_line_meta</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="c1"># If the specified cell line type can be found in the database,</span>
            <span class="c1"># we can compare these keys and fetch the corresponding metadata.</span>

            <span class="n">not_matched_identifiers</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">cell_line_meta</span><span class="p">[</span><span class="n">identifier_type</span><span class="p">]))</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">not_matched_identifiers</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
                <span class="nb">print</span><span class="p">(</span>
                    <span class="s2">&quot;Following identifiers can not be found in cell line annotation file,&quot;</span>
                    <span class="s2">&quot; so their corresponding meta data are NA values. Please check it again:&quot;</span><span class="p">,</span>
                    <span class="o">*</span><span class="n">not_matched_identifiers</span><span class="p">,</span>
                    <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
                <span class="p">)</span>

            <span class="k">if</span> <span class="n">cell_line_information</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># If no cell_line_information is specified, all metadata is fetched by default.</span>
                <span class="c1"># Sometimes there is already different cell line information in the `adata`.</span>
                <span class="c1"># To avoid redundant information we will remove the duplicate information from metadata after merging.</span>
                <span class="n">adata</span><span class="o">.</span><span class="n">obs</span> <span class="o">=</span> <span class="p">(</span>
                    <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
                    <span class="o">.</span><span class="n">merge</span><span class="p">(</span>
                        <span class="n">cell_line_meta</span><span class="p">,</span>
                        <span class="n">left_on</span><span class="o">=</span><span class="n">cell_line_identifier</span><span class="p">,</span>
                        <span class="n">right_on</span><span class="o">=</span><span class="n">identifier_type</span><span class="p">,</span>
                        <span class="n">how</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">,</span>
                        <span class="n">suffixes</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="s2">&quot;_fromMeta&quot;</span><span class="p">),</span>
                    <span class="p">)</span>
                    <span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s2">&quot;^(?!.*_fromMeta)&quot;</span><span class="p">)</span>
                    <span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">names</span><span class="p">)</span>
                <span class="p">)</span>
                <span class="c1"># If cell_line_identifier and identifier_type have different names,</span>
                <span class="c1"># there will be a column for each of them after merging,</span>
                <span class="c1"># which is redundant as they refer to the same information.</span>
                <span class="c1"># We will move the identifier_type column.</span>
                <span class="k">if</span> <span class="n">cell_line_identifier</span> <span class="o">!=</span> <span class="n">identifier_type</span><span class="p">:</span>
                    <span class="k">del</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">identifier_type</span><span class="p">]</span>

            <span class="k">elif</span> <span class="nb">set</span><span class="p">(</span><span class="n">cell_line_information</span><span class="p">)</span><span class="o">.</span><span class="n">issubset</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">cell_line_meta</span><span class="o">.</span><span class="n">columns</span><span class="p">)):</span>
                <span class="c1"># If cell_line_information is specified and can be found in the DepMap database,</span>
                <span class="c1"># We will subset the original metadata dataframe correspondingly and add them to the `adata`.</span>
                <span class="c1"># Again, redundant information will be removed.</span>
                <span class="k">if</span> <span class="n">identifier_type</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">cell_line_information</span><span class="p">:</span>
                    <span class="n">cell_line_information</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">identifier_type</span><span class="p">)</span>
                <span class="n">cell_line_meta_part</span> <span class="o">=</span> <span class="n">cell_line_meta</span><span class="p">[</span><span class="n">cell_line_information</span><span class="p">]</span>
                <span class="n">adata</span><span class="o">.</span><span class="n">obs</span> <span class="o">=</span> <span class="p">(</span>
                    <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
                    <span class="o">.</span><span class="n">merge</span><span class="p">(</span>
                        <span class="n">cell_line_meta_part</span><span class="p">,</span>
                        <span class="n">left_on</span><span class="o">=</span><span class="n">cell_line_identifier</span><span class="p">,</span>
                        <span class="n">right_on</span><span class="o">=</span><span class="n">identifier_type</span><span class="p">,</span>
                        <span class="n">how</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">,</span>
                        <span class="n">suffixes</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="s2">&quot;_fromMeta&quot;</span><span class="p">),</span>
                    <span class="p">)</span>
                    <span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s2">&quot;^(?!.*_fromMeta)&quot;</span><span class="p">)</span>
                    <span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">names</span><span class="p">)</span>
                <span class="p">)</span>
                <span class="c1"># Again, redundant information will be removed.</span>
                <span class="k">if</span> <span class="n">cell_line_identifier</span> <span class="o">!=</span> <span class="n">identifier_type</span><span class="p">:</span>
                    <span class="k">del</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">identifier_type</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                    <span class="s2">&quot;The specified cell line metadata can&#39;t be found in the DepMap database.&quot;</span>
                    <span class="s2">&quot;Please give the cell line metadata available in the DepMap,&quot;</span>
                    <span class="s2">&quot;or fetch all the metadata by default.&quot;</span>
                <span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified cell line type is not available in the DepMap database.&quot;</span>
                <span class="s2">&quot;Please give the type of cell line information available in the DepMap.&quot;</span>
                <span class="s2">&quot;e.g. DepMap_ID, cell_line_name or stripped_cell_line_name.&quot;</span>
                <span class="s2">&quot;DepMap_ID is compared by default.&quot;</span>
            <span class="p">)</span>

        <span class="k">return</span> <span class="n">adata</span></div>

<div class="viewcode-block" id="CellLineMetaData.annotate_bulk_rna_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.annotate_bulk_rna_expression">[docs]</a>    <span class="k">def</span> <span class="nf">annotate_bulk_rna_expression</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">adata</span><span class="p">:</span> <span class="n">AnnData</span><span class="p">,</span>
        <span class="n">cell_line_identifier</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;cell_line_name&quot;</span><span class="p">,</span>
        <span class="n">identifier_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;model_name&quot;</span><span class="p">,</span>
        <span class="n">bulk_rna_source</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;broad&quot;</span><span class="p">,</span>
        <span class="n">bulk_rna_information</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;read_count&quot;</span><span class="p">,</span>
        <span class="n">copy</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Fetch bulk rna expression.</span>

<span class="sd">        For each cell, we fetch bulk rna expression from either Broad or Sanger cell line.</span>

<span class="sd">        Args:</span>
<span class="sd">            adata: The data object to annotate.</span>
<span class="sd">            cell_line_identifier: The column of `.obs` with cell line information. Defaults to &quot;cell_line_name&quot;.</span>
<span class="sd">            identifier_type: The type of cell line information, e.g. model_name or model_id. Defaults to &quot;model_name&quot;.</span>
<span class="sd">            bulk_rna_source: The bulk rna expression data from either Broad or Sanger cell line. Defaults to &quot;broad&quot;.</span>
<span class="sd">            bulk_rna_information: The metadata to fetch. Defaults to &quot;read_count&quot;.</span>
<span class="sd">            copy: Determines whether a copy of the `adata` is returned. Defaults to False.</span>

<span class="sd">        Returns:</span>
<span class="sd">            Returns an AnnData object with bulk rna expression annotation.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">copy</span><span class="p">:</span>
            <span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
        <span class="k">if</span> <span class="n">bulk_rna_source</span> <span class="o">==</span> <span class="s2">&quot;broad&quot;</span><span class="p">:</span>
            <span class="n">bulk_rna</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_broad</span>
        <span class="k">elif</span> <span class="n">bulk_rna_source</span> <span class="o">==</span> <span class="s2">&quot;sanger&quot;</span><span class="p">:</span>
            <span class="n">bulk_rna</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">bulk_rna_sanger</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified source of bulk rna expression data is not available. &quot;</span>
                <span class="s2">&quot;Please choose either broad or sanger.&quot;</span>
            <span class="p">)</span>

        <span class="c1"># Make sure that the specified `cell_line_type` can be found in the bulk rna expression data,</span>
        <span class="c1"># then we can compare these keys and fetch the corresponding metadata.</span>
        <span class="k">if</span> <span class="n">cell_line_identifier</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified `cell_line_identifier` can&#39;t be found in the `adata.obs`. &quot;</span>
                <span class="s2">&quot;Please fetch the cell line metadata first using the function &quot;</span>
                <span class="s2">&quot;`annotate_cell_lines()`.&quot;</span>
            <span class="p">)</span>

        <span class="k">if</span> <span class="n">identifier_type</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">bulk_rna</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified `identifier_type` can&#39;t be found in the metadata. &quot;</span>
                <span class="s2">&quot;Please check the available `identifier_type` in the bulk expression data using &quot;</span>
                <span class="s2">&quot;the function getinfo_annotate_bulk_rna_expression().&quot;</span>
            <span class="p">)</span>

        <span class="n">not_matched_identifiers</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">bulk_rna</span><span class="p">[</span><span class="n">identifier_type</span><span class="p">]))</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">not_matched_identifiers</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]Following identifiers can not be found in bulk RNA expression data,&quot;</span>
                <span class="s2">&quot; their corresponding meta data are NA values. Please check it again:&quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="n">not_matched_identifiers</span><span class="p">,</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>

        <span class="n">rna_exp</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">pivot</span><span class="p">(</span><span class="n">bulk_rna</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="n">identifier_type</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="s2">&quot;gene_id&quot;</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="n">bulk_rna_information</span><span class="p">)</span>
        <span class="n">rna_exp</span> <span class="o">=</span> <span class="n">rna_exp</span><span class="o">.</span><span class="n">reindex</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span>
        <span class="n">rna_exp</span> <span class="o">=</span> <span class="n">rna_exp</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
        <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s2">&quot;bulk_rna_expression_&quot;</span> <span class="o">+</span> <span class="n">bulk_rna_source</span><span class="p">]</span> <span class="o">=</span> <span class="n">rna_exp</span>

        <span class="k">return</span> <span class="n">adata</span></div>

<div class="viewcode-block" id="CellLineMetaData.annotate_protein_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.annotate_protein_expression">[docs]</a>    <span class="k">def</span> <span class="nf">annotate_protein_expression</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">adata</span><span class="p">:</span> <span class="n">AnnData</span><span class="p">,</span>
        <span class="n">cell_line_identifier</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;cell_line_name&quot;</span><span class="p">,</span>
        <span class="n">identifier_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;model_name&quot;</span><span class="p">,</span>
        <span class="n">copy</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Fetch protein expression.</span>

<span class="sd">        For each cell, we fetch protein intensity values acquired using data-independent acquisition mass spectrometry (DIA-MS).</span>

<span class="sd">        Args:</span>
<span class="sd">            adata: The data object to annotate.</span>
<span class="sd">            cell_line_identifier: The column of `.obs` with cell line information. Defaults to &quot;cell_line_name&quot;.</span>
<span class="sd">            identifier_type: The type of cell line information, e.g. model_name or model_id. Defaults to &quot;model_name&quot;.</span>
<span class="sd">            copy: Determines whether a copy of the `adata` is returned. Defaults to False.</span>

<span class="sd">        Returns:</span>
<span class="sd">            Returns an AnnData object with protein expression annotation.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">copy</span><span class="p">:</span>
            <span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>

        <span class="c1"># Make sure that the specified `cell_line_type` can be found in the protein expression data,</span>
        <span class="c1"># then we can compare these keys and fetch the corresponding metadata.</span>
        <span class="k">if</span> <span class="n">cell_line_identifier</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified `cell_line_identifier` can&#39;t be found in the `adata.obs`. &quot;</span>
                <span class="s2">&quot;Please fetch the cell line meta data first using the function &quot;</span>
                <span class="s2">&quot;`annotate_cell_lines()`.&quot;</span>
            <span class="p">)</span>

        <span class="k">if</span> <span class="n">identifier_type</span> <span class="ow">not</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified `identifier_type` can&#39;t be found in the meta data. &quot;</span>
                <span class="s2">&quot;Please check the available identifier types in the protein expression data calling &quot;</span>
                <span class="s2">&quot;the function `getinfo_annotate_protein_expression()`.&quot;</span>
            <span class="p">)</span>

        <span class="n">not_matched_identifiers</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span>
            <span class="nb">set</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="p">[</span><span class="n">identifier_type</span><span class="p">])</span>
        <span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">not_matched_identifiers</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]Following identifiers can not be found in the protein expression data,&quot;</span>
                <span class="s2">&quot; their corresponding meta data are NA values. Please check it again:&quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="n">not_matched_identifiers</span><span class="p">,</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="c1"># convert the original protein intensities table from long format to wide format, group by the cell lines</span>
        <span class="n">prot_exp</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">pivot</span><span class="p">(</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">proteomics_data</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="n">identifier_type</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="s2">&quot;uniprot_id&quot;</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="s2">&quot;protein_intensity&quot;</span>
        <span class="p">)</span>
        <span class="n">prot_exp</span> <span class="o">=</span> <span class="n">prot_exp</span><span class="o">.</span><span class="n">reindex</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span>
        <span class="n">prot_exp</span> <span class="o">=</span> <span class="n">prot_exp</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
        <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s2">&quot;proteomics_protein_intensity&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">prot_exp</span>

        <span class="k">return</span> <span class="n">adata</span></div>

<div class="viewcode-block" id="CellLineMetaData.annotate_ccle_expression"><a class="viewcode-back" href="../../../../usage/tools/pertpy.tools.CellLineMetaData.html#pertpy.tools.CellLineMetaData.annotate_ccle_expression">[docs]</a>    <span class="k">def</span> <span class="nf">annotate_ccle_expression</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">adata</span><span class="p">:</span> <span class="n">AnnData</span><span class="p">,</span>
        <span class="n">cell_line_identifier</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;DepMap_ID&quot;</span><span class="p">,</span>
        <span class="n">copy</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">AnnData</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Fetch CCLE expression data.</span>

<span class="sd">         For each cell, we fetch gene expression TPM values of the protein coding genes for its corresponding DepMap</span>
<span class="sd">         cell line.</span>

<span class="sd">        Args:</span>
<span class="sd">            adata: The data object to annotate.</span>
<span class="sd">            cell_line_identifier: The column of `.obs` with cell line information. Defaults to &quot;DepMap_ID&quot;.</span>
<span class="sd">            copy: Determines whether a copy of the `adata` is returned. Defaults to False.</span>

<span class="sd">        Returns:</span>
<span class="sd">            Returns an AnnData object with CCLE expression annotation.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">copy</span><span class="p">:</span>
            <span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>

        <span class="c1"># Make sure that the specified cell line type can be found in the CCLE expression data,</span>
        <span class="c1"># then we can compare these keys and fetch the corresponding metadata.</span>
        <span class="k">if</span> <span class="n">cell_line_identifier</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="s2">&quot;The specified `cell_line_identifier` can&#39;t be found in the `adata.obs`. &quot;</span>
                <span class="s2">&quot;Please fetch the cell line meta data first using the function &quot;</span>
                <span class="s2">&quot;`annotate_cell_lines()`.&quot;</span>
            <span class="p">)</span>

        <span class="n">not_matched_identifiers</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ccle_expr</span><span class="o">.</span><span class="n">index</span><span class="p">))</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">not_matched_identifiers</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="nb">print</span><span class="p">(</span>
                <span class="s2">&quot;[bold yellow]Following identifiers can not be found in the CCLE expression data,&quot;</span>
                <span class="s2">&quot; their corresponding meta data are NA values. Please check it again:&quot;</span><span class="p">,</span>
                <span class="o">*</span><span class="n">not_matched_identifiers</span><span class="p">,</span>
                <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">- &quot;</span><span class="p">,</span>
            <span class="p">)</span>

        <span class="n">ccle_expression</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ccle_expr</span><span class="o">.</span><span class="n">reindex</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="n">cell_line_identifier</span><span class="p">])</span>
        <span class="n">ccle_expression</span> <span class="o">=</span> <span class="n">ccle_expression</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
        <span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s2">&quot;CCLE_expression&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">ccle_expression</span>

        <span class="k">return</span> <span class="n">adata</span></div></div>
</pre></div>
        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2021, Lukas Heumos, Theislab
            </div>
            Made with 
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer no-toc">
      
      
      
    </aside>
  </div>
</div><script data-url_root="../../../../" id="documentation_options" src="../../../../_static/documentation_options.js?v=3fadbb4a"></script>
    <script src="../../../../_static/doctools.js?v=888ff710"></script>
    <script src="../../../../_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="../../../../_static/scripts/furo.js?v=32e29ea5"></script>
    <script src="../../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../../_static/design-tabs.js?v=36754332"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>